<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Clarke Iakovakis" />


<title>rcrossref</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/jsoneditor-5.25.6/jsoneditor.min.css" rel="stylesheet" />
<script src="site_libs/jsoneditor-5.25.6/jsoneditor.min.js"></script>
<script src="site_libs/jsonedit-binding-3.0.0/jsonedit.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 52px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h2 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h3 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h4 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h5 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h6 {
  padding-top: 57px;
  margin-top: -57px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Clarke Iakovakis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    FSCI 2020 W24 Session
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="fsci_syllabus.html">Course syllabus</a>
    </li>
    <li>
      <a href="using_jupyter_notebooks.html">Jupyter Notebooks and Binder</a>
    </li>
    <li>
      <a href="IntroductionToR.html">Introduction to R</a>
    </li>
    <li>
      <a href="RBasics.html">R Basics</a>
    </li>
    <li>
      <a href="DataExploration.html">Data Exploration</a>
    </li>
    <li>
      <a href="jennybc_lists_lesson.html">Exploring Data in Lists</a>
    </li>
    <li>
      <a href="rcrossref.html">rcrossref</a>
    </li>
    <li>
      <a href="rorcid.html">rorcid</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">rcrossref</h1>
<h4 class="author">Clarke Iakovakis</h4>

</div>


<div id="licensing" class="section level1">
<h1>Licensing</h1>
<p>This walkthrough is distributed under a <a href="Creative%20Commons%20Attribution%204.0%20International%20(CC%20BY%204.0)%20License">https://creativecommons.org/licenses/by/4.0/</a>.</p>
</div>
<div id="binder-link-to-this-notebook" class="section level1">
<h1>Binder link to this notebook:</h1>
<p><a href="https://mybinder.org/v2/gh/ciakovx/ciakovx.github.io/master?filepath=rcrossref.ipynb"><img src="https://mybinder.org/badge_logo.svg" alt="Binder" /></a></p>
<div id="setting-up-rcrossref" class="section level2">
<h2>Setting up <code>rcrossref</code></h2>
<p>First install and load <code>rcrossref</code> in R. Also install <code>usethis</code> in order to send your email to Crossref, as well as <code>tidyverse</code>, which is a package that contains many packages we’ll use throughout.</p>
<pre class="r"><code>install.packages(&quot;rcrossref&quot;)
install.packages(&quot;usethis&quot;)
install.packages(&quot;tidyverse&quot;)
install.packages(&quot;listviewer&quot;)
library(rcrossref)
library(usethis)
library(tidyverse)
library(listviewer)</code></pre>
<p>As described in the documentation, the Crossref team encourages requests with appropriate contact information, and will forward you to a dedicated API cluster for improved performance when you share your email with them. Learn more at <a href="https://github.com/CrossRef/rest-api-doc#good-manners--more-reliable-service" class="uri">https://github.com/CrossRef/rest-api-doc#good-manners--more-reliable-service</a>. To do so, first open your R environment using the <code>edit_r_environ()</code> function from the <code>usethis</code> package.</p>
<pre class="r"><code>usethis::edit_r_environ()</code></pre>
<p>A new window will open in R Studio. Type <code>crossref_email=name@example.com</code>, replacing the example with your own email address. Then press enter to create a new line, and leave it blank. Press Ctrl/Cmd + S to save the API key to your R environment and close the window. Your email address will now be shared with Crossref whenever you send them an API request.</p>
</div>
</div>
<div id="crossref-and-rcrossref" class="section level1">
<h1>Crossref and <code>rcrossref</code></h1>
<div id="crossref" class="section level2">
<h2>Crossref</h2>
<p>Crossref is a a not-for-profit membership organization dedicated to interlinking scholarly metadata, including journals, books, conference proceedings, working papers, technical reports, data sets, authors, funders, and more. The <a href="https://github.com/CrossRef/rest-api-doc">Crossref REST API</a> allows anybody to search and reuse members’ metadata in a variety of ways. Read <a href="https://www.crossref.org/services/metadata-delivery/user-stories/">examples of user stories</a>.</p>
</div>
<div id="rcrossref" class="section level2">
<h2><code>rcrossref</code></h2>
<p><code>rcrossref</code> is a package developed by <a href="https://scottchamberlain.info/">Scott Chamberlain</a>, Hao Zhu, Najko Jahn, Carl Boettiger, and Karthik Ram, part of the <a href="https://ropensci.org/">rOpenSci</a> set of packages. rOpenSci is an incredible organization dedicated to open and reproducible research using shared data and reusable software. I strongly recommend you browse their set of packages at <a href="https://ropensci.org/packages/" class="uri">https://ropensci.org/packages/</a>.</p>
<p><code>rcrossref</code> serves as an interface to the Crossref API.</p>
<p><strong>Key links</strong></p>
<ul>
<li><a href="https://cran.r-project.org/web/packages/rcrossref/rcrossref.pdf">rcrossref documentation</a></li>
<li><a href="https://github.com/ropensci/rcrossref">Crossref REST API documentation</a></li>
<li><a href="https://github.com/CrossRef/rest-api-doc/blob/master/api_format.md">Crossref Metadata API JSON Format</a></li>
<li>Crossref and rcrossref tutorials
<ul>
<li><a href="https://ciakovx.github.io/rcrossref.html">my <code>rcrossref</code> tutorial</a></li>
<li>rOpenSci <a href="https://ropensci.org/tutorials/rcrossref_tutorial/">rcrossref tutorial</a></li>
<li>Paul Oldham: <a href="https://poldham.github.io/abs/crossref.html">“Accessing the Scientific Literature with Crossref”</a></li>
<li><a href="https://cran.r-project.org/web/packages/rcrossref/vignettes/crossref_vignette.html">rcrossref vignette</a></li>
</ul></li>
</ul>
</div>
</div>
<div id="getting-publications-from-journals-with-cr_journals" class="section level1">
<h1>Getting publications from journals with <code>cr_journals</code></h1>
<p><code>cr_journals()</code> takes either an ISSN or a general keyword query, and returns metadata for articles published in the journal, including DOI, title, volume, issue, pages, publisher, authors, etc. A full list of publications in Crossref is <a href="https://support.crossref.org/hc/en-us/articles/213197226-Browsable-title-list">available on their website</a>.</p>
<div id="getting-journal-details" class="section level2">
<h2>Getting journal details</h2>
<p>Crossref is entirely dependent on publishers to supply the metadata. Some fields are required, while others are optional. You may therefore first be interested in what metadata publishers have submitted to Crossref for a given journal. By using <code>cr_journals</code> with <code>works = FALSE</code>, you can determine who publishes the journal, the total number of articles for the journal in Crossref, whether abstracts are included, if the full text of articles is deposited, if author ORCIDs are provided, and if the publisher supplies author affiliations, author ORCID iDs, article licensing data, funders for the article, article references, and a few other items.</p>
<p>First we will create a new vector <code>plosone_issn</code> with the ISSN for the journal <em>PLoS ONE</em>.</p>
<pre class="r"><code># assign the PLoS ISSN
plosone_issn &lt;- &quot;1932-6203&quot;</code></pre>
<p>We will then run <code>rcrossref::cr_journals()</code>, setting the ISSN equal to the <code>plosone_issn</code> we just created, and print the results.</p>
<pre class="r"><code># get information about the journal
plosone_details &lt;- rcrossref::cr_journals(issn = plosone_issn, works = FALSE)
plosone_details</code></pre>
<pre><code>## $meta
## NULL
## 
## $data
##      title                 publisher      issn last_status_check_time
## 1 PLoS ONE Public Library of Science 1932-6203             2020-07-23
##   deposits_abstracts_current deposits_orcids_current deposits
## 1                      FALSE                    TRUE     TRUE
##   deposits_affiliations_backfile deposits_update_policies_backfile
## 1                          FALSE                              TRUE
##   deposits_similarity_checking_backfile deposits_award_numbers_current
## 1                                  TRUE                           TRUE
##   deposits_resource_links_current deposits_articles
## 1                           FALSE              TRUE
##   deposits_affiliations_current deposits_funders_current
## 1                         FALSE                     TRUE
##   deposits_references_backfile deposits_abstracts_backfile
## 1                         TRUE                       FALSE
##   deposits_licenses_backfile deposits_award_numbers_backfile
## 1                       TRUE                            TRUE
##   deposits_open_references_backfile deposits_open_references_current
## 1                              TRUE                             TRUE
##   deposits_references_current deposits_resource_links_backfile
## 1                        TRUE                            FALSE
##   deposits_orcids_backfile deposits_funders_backfile
## 1                     TRUE                      TRUE
##   deposits_update_policies_current deposits_similarity_checking_current
## 1                             TRUE                                 TRUE
##   deposits_licenses_current affiliations_current similarity_checking_current
## 1                      TRUE                    0                           1
##   funders_backfile licenses_backfile funders_current affiliations_backfile
## 1        0.1275025         0.9917417       0.5907303                     0
##   resource_links_backfile orcids_backfile update_policies_current
## 1                       0       0.1000805                       1
##   open_references_backfile orcids_current similarity_checking_backfile
## 1                        1      0.9235606                     0.905079
##   references_backfile award_numbers_backfile update_policies_backfile
## 1            0.905079              0.1087021                0.9999481
##   licenses_current award_numbers_current abstracts_backfile
## 1                1               0.51173                  0
##   resource_links_current abstracts_current open_references_current
## 1                      0                 0                       1
##   references_current total_dois current_dois backfile_dois
## 1                  1     236389        43734        192655
## 
## $facets
## NULL</code></pre>
<p>This actually comes back as a list of three items: <code>meta</code>, <code>data</code>, and <code>facets</code>. The good stuff is in <code>data</code>.</p>
<p>We use the <code>pluck()</code> function from the <code>purrr</code> package to pull that data only. We will be using <code>pluck</code> throughout this tutorial; it’s an easy way of indexing deeply and flexibly into lists to extract information.</p>
<p>We don’t have time in this tutorial to discuss list items and purr. For an excellent in-depth tutorial, see Jenny Bryan’s <a href="https://jennybc.github.io/purrr-tutorial/ls01_map-name-position-shortcuts.html">Introduction to map(): extract elements</a>, also provided on this website.</p>
<pre class="r"><code># get information about the journal and pluck the data
plosone_details &lt;- rcrossref::cr_journals(issn = plosone_issn, works = FALSE) %&gt;% 
    purrr::pluck(&quot;data&quot;)</code></pre>
<p>The <code>purrr::pluck()</code> function is connected to <code>plosone_details</code> with something called a <a href="https://www.datacamp.com/community/tutorials/pipe-r-tutorial">Pipe Operator</a> <code>%&amp;gt;%</code>, which we will be using throughout the tutorial. A pipe takes the output of one statement and immediately makes it the input of the next statement. It helps so that you don’t have to write every intermediate, processing data to your R environment. You can think of it as “then” in natural language. So the above script first makes the API call with <code>cr_journals()</code>, then it applies <code>pluck()</code> to extract only the list element called <code>&quot;data&quot;</code>, and returns it to the <code>plosone_details</code> value.</p>
<p>We now have a <strong>data frame</strong> including the details Croassref has on file about PLoS ONE. Scroll to the right to see all the columns.</p>
<pre class="r"><code>plosone_details</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["title"],"name":[1],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[2],"type":["chr"],"align":["left"]},{"label":["issn"],"name":[3],"type":["chr"],"align":["left"]},{"label":["last_status_check_time"],"name":[4],"type":["date"],"align":["right"]},{"label":["deposits_abstracts_current"],"name":[5],"type":["lgl"],"align":["right"]},{"label":["deposits_orcids_current"],"name":[6],"type":["lgl"],"align":["right"]},{"label":["deposits"],"name":[7],"type":["lgl"],"align":["right"]},{"label":["deposits_affiliations_backfile"],"name":[8],"type":["lgl"],"align":["right"]},{"label":["deposits_update_policies_backfile"],"name":[9],"type":["lgl"],"align":["right"]},{"label":["deposits_similarity_checking_backfile"],"name":[10],"type":["lgl"],"align":["right"]},{"label":["deposits_award_numbers_current"],"name":[11],"type":["lgl"],"align":["right"]},{"label":["deposits_resource_links_current"],"name":[12],"type":["lgl"],"align":["right"]},{"label":["deposits_articles"],"name":[13],"type":["lgl"],"align":["right"]},{"label":["deposits_affiliations_current"],"name":[14],"type":["lgl"],"align":["right"]},{"label":["deposits_funders_current"],"name":[15],"type":["lgl"],"align":["right"]},{"label":["deposits_references_backfile"],"name":[16],"type":["lgl"],"align":["right"]},{"label":["deposits_abstracts_backfile"],"name":[17],"type":["lgl"],"align":["right"]},{"label":["deposits_licenses_backfile"],"name":[18],"type":["lgl"],"align":["right"]},{"label":["deposits_award_numbers_backfile"],"name":[19],"type":["lgl"],"align":["right"]},{"label":["deposits_open_references_backfile"],"name":[20],"type":["lgl"],"align":["right"]},{"label":["deposits_open_references_current"],"name":[21],"type":["lgl"],"align":["right"]},{"label":["deposits_references_current"],"name":[22],"type":["lgl"],"align":["right"]},{"label":["deposits_resource_links_backfile"],"name":[23],"type":["lgl"],"align":["right"]},{"label":["deposits_orcids_backfile"],"name":[24],"type":["lgl"],"align":["right"]},{"label":["deposits_funders_backfile"],"name":[25],"type":["lgl"],"align":["right"]},{"label":["deposits_update_policies_current"],"name":[26],"type":["lgl"],"align":["right"]},{"label":["deposits_similarity_checking_current"],"name":[27],"type":["lgl"],"align":["right"]},{"label":["deposits_licenses_current"],"name":[28],"type":["lgl"],"align":["right"]},{"label":["affiliations_current"],"name":[29],"type":["dbl"],"align":["right"]},{"label":["similarity_checking_current"],"name":[30],"type":["dbl"],"align":["right"]},{"label":["funders_backfile"],"name":[31],"type":["dbl"],"align":["right"]},{"label":["licenses_backfile"],"name":[32],"type":["dbl"],"align":["right"]},{"label":["funders_current"],"name":[33],"type":["dbl"],"align":["right"]},{"label":["affiliations_backfile"],"name":[34],"type":["dbl"],"align":["right"]},{"label":["resource_links_backfile"],"name":[35],"type":["dbl"],"align":["right"]},{"label":["orcids_backfile"],"name":[36],"type":["dbl"],"align":["right"]},{"label":["update_policies_current"],"name":[37],"type":["dbl"],"align":["right"]},{"label":["open_references_backfile"],"name":[38],"type":["dbl"],"align":["right"]},{"label":["orcids_current"],"name":[39],"type":["dbl"],"align":["right"]},{"label":["similarity_checking_backfile"],"name":[40],"type":["dbl"],"align":["right"]},{"label":["references_backfile"],"name":[41],"type":["dbl"],"align":["right"]},{"label":["award_numbers_backfile"],"name":[42],"type":["dbl"],"align":["right"]},{"label":["update_policies_backfile"],"name":[43],"type":["dbl"],"align":["right"]},{"label":["licenses_current"],"name":[44],"type":["dbl"],"align":["right"]},{"label":["award_numbers_current"],"name":[45],"type":["dbl"],"align":["right"]},{"label":["abstracts_backfile"],"name":[46],"type":["dbl"],"align":["right"]},{"label":["resource_links_current"],"name":[47],"type":["dbl"],"align":["right"]},{"label":["abstracts_current"],"name":[48],"type":["dbl"],"align":["right"]},{"label":["open_references_current"],"name":[49],"type":["dbl"],"align":["right"]},{"label":["references_current"],"name":[50],"type":["dbl"],"align":["right"]},{"label":["total_dois"],"name":[51],"type":["int"],"align":["right"]},{"label":["current_dois"],"name":[52],"type":["int"],"align":["right"]},{"label":["backfile_dois"],"name":[53],"type":["int"],"align":["right"]}],"data":[{"1":"PLoS ONE","2":"Public Library of Science","3":"1932-6203","4":"2020-07-23","5":"FALSE","6":"TRUE","7":"TRUE","8":"FALSE","9":"TRUE","10":"TRUE","11":"TRUE","12":"FALSE","13":"TRUE","14":"FALSE","15":"TRUE","16":"TRUE","17":"FALSE","18":"TRUE","19":"TRUE","20":"TRUE","21":"TRUE","22":"TRUE","23":"FALSE","24":"TRUE","25":"TRUE","26":"TRUE","27":"TRUE","28":"TRUE","29":"0","30":"1","31":"0.1275025","32":"0.9917417","33":"0.5907303","34":"0","35":"0","36":"0.1000805","37":"1","38":"1","39":"0.9235606","40":"0.905079","41":"0.905079","42":"0.1087021","43":"0.9999481","44":"1","45":"0.51173","46":"0","47":"0","48":"0","49":"1","50":"1","51":"236389","52":"43734","53":"192655"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>There are a number of ways to explore this data frame:</p>
<pre class="r"><code># display information about the data frame
str(plosone_details)</code></pre>
<pre><code>## &#39;data.frame&#39;:    1 obs. of  53 variables:
##  $ title                                : chr &quot;PLoS ONE&quot;
##  $ publisher                            : chr &quot;Public Library of Science&quot;
##  $ issn                                 : chr &quot;1932-6203&quot;
##  $ last_status_check_time               : Date, format: &quot;2020-07-23&quot;
##  $ deposits_abstracts_current           : logi FALSE
##  $ deposits_orcids_current              : logi TRUE
##  $ deposits                             : logi TRUE
##  $ deposits_affiliations_backfile       : logi FALSE
##  $ deposits_update_policies_backfile    : logi TRUE
##  $ deposits_similarity_checking_backfile: logi TRUE
##  $ deposits_award_numbers_current       : logi TRUE
##  $ deposits_resource_links_current      : logi FALSE
##  $ deposits_articles                    : logi TRUE
##  $ deposits_affiliations_current        : logi FALSE
##  $ deposits_funders_current             : logi TRUE
##  $ deposits_references_backfile         : logi TRUE
##  $ deposits_abstracts_backfile          : logi FALSE
##  $ deposits_licenses_backfile           : logi TRUE
##  $ deposits_award_numbers_backfile      : logi TRUE
##  $ deposits_open_references_backfile    : logi TRUE
##  $ deposits_open_references_current     : logi TRUE
##  $ deposits_references_current          : logi TRUE
##  $ deposits_resource_links_backfile     : logi FALSE
##  $ deposits_orcids_backfile             : logi TRUE
##  $ deposits_funders_backfile            : logi TRUE
##  $ deposits_update_policies_current     : logi TRUE
##  $ deposits_similarity_checking_current : logi TRUE
##  $ deposits_licenses_current            : logi TRUE
##  $ affiliations_current                 : num 0
##  $ similarity_checking_current          : num 1
##  $ funders_backfile                     : num 0.128
##  $ licenses_backfile                    : num 0.992
##  $ funders_current                      : num 0.591
##  $ affiliations_backfile                : num 0
##  $ resource_links_backfile              : num 0
##  $ orcids_backfile                      : num 0.1
##  $ update_policies_current              : num 1
##  $ open_references_backfile             : num 1
##  $ orcids_current                       : num 0.924
##  $ similarity_checking_backfile         : num 0.905
##  $ references_backfile                  : num 0.905
##  $ award_numbers_backfile               : num 0.109
##  $ update_policies_backfile             : num 1
##  $ licenses_current                     : num 1
##  $ award_numbers_current                : num 0.512
##  $ abstracts_backfile                   : num 0
##  $ resource_links_current               : num 0
##  $ abstracts_current                    : num 0
##  $ open_references_current              : num 1
##  $ references_current                   : num 1
##  $ total_dois                           : int 236389
##  $ current_dois                         : int 43734
##  $ backfile_dois                        : int 192655</code></pre>
<p>Type <code>?str</code> into the console to read the description of the <code>str</code> function. You can call <code>str()</code> on an R object to compactly display information about it, including the data type, the number of elements, and a printout of the first few elements.</p>
<pre class="r"><code># dimensions: 1 row, 53 columns
dim(plosone_details)</code></pre>
<pre><code>## [1]  1 53</code></pre>
<pre class="r"><code># number of rows
nrow(plosone_details)</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code># number of columns
ncol(plosone_details)</code></pre>
<pre><code>## [1] 53</code></pre>
<pre class="r"><code># column names
names(plosone_details)</code></pre>
<pre><code>##  [1] &quot;title&quot;                                
##  [2] &quot;publisher&quot;                            
##  [3] &quot;issn&quot;                                 
##  [4] &quot;last_status_check_time&quot;               
##  [5] &quot;deposits_abstracts_current&quot;           
##  [6] &quot;deposits_orcids_current&quot;              
##  [7] &quot;deposits&quot;                             
##  [8] &quot;deposits_affiliations_backfile&quot;       
##  [9] &quot;deposits_update_policies_backfile&quot;    
## [10] &quot;deposits_similarity_checking_backfile&quot;
## [11] &quot;deposits_award_numbers_current&quot;       
## [12] &quot;deposits_resource_links_current&quot;      
## [13] &quot;deposits_articles&quot;                    
## [14] &quot;deposits_affiliations_current&quot;        
## [15] &quot;deposits_funders_current&quot;             
## [16] &quot;deposits_references_backfile&quot;         
## [17] &quot;deposits_abstracts_backfile&quot;          
## [18] &quot;deposits_licenses_backfile&quot;           
## [19] &quot;deposits_award_numbers_backfile&quot;      
## [20] &quot;deposits_open_references_backfile&quot;    
## [21] &quot;deposits_open_references_current&quot;     
## [22] &quot;deposits_references_current&quot;          
## [23] &quot;deposits_resource_links_backfile&quot;     
## [24] &quot;deposits_orcids_backfile&quot;             
## [25] &quot;deposits_funders_backfile&quot;            
## [26] &quot;deposits_update_policies_current&quot;     
## [27] &quot;deposits_similarity_checking_current&quot; 
## [28] &quot;deposits_licenses_current&quot;            
## [29] &quot;affiliations_current&quot;                 
## [30] &quot;similarity_checking_current&quot;          
## [31] &quot;funders_backfile&quot;                     
## [32] &quot;licenses_backfile&quot;                    
## [33] &quot;funders_current&quot;                      
## [34] &quot;affiliations_backfile&quot;                
## [35] &quot;resource_links_backfile&quot;              
## [36] &quot;orcids_backfile&quot;                      
## [37] &quot;update_policies_current&quot;              
## [38] &quot;open_references_backfile&quot;             
## [39] &quot;orcids_current&quot;                       
## [40] &quot;similarity_checking_backfile&quot;         
## [41] &quot;references_backfile&quot;                  
## [42] &quot;award_numbers_backfile&quot;               
## [43] &quot;update_policies_backfile&quot;             
## [44] &quot;licenses_current&quot;                     
## [45] &quot;award_numbers_current&quot;                
## [46] &quot;abstracts_backfile&quot;                   
## [47] &quot;resource_links_current&quot;               
## [48] &quot;abstracts_current&quot;                    
## [49] &quot;open_references_current&quot;              
## [50] &quot;references_current&quot;                   
## [51] &quot;total_dois&quot;                           
## [52] &quot;current_dois&quot;                         
## [53] &quot;backfile_dois&quot;</code></pre>
<p>We see this data frame includes one observation of 53 different variables. This includes the total number of DOIs, whether the abstracts, orcids, article references are current; and other information.</p>
<p>You can use the $ symbol to work with particular variables. For example, the <code>publisher</code> column:</p>
<pre class="r"><code># print the publisher variable
plosone_details$publisher</code></pre>
<pre><code>## [1] &quot;Public Library of Science&quot;</code></pre>
<p>The total number of DOIs on file:</p>
<pre class="r"><code># print the total number of DOIs
plosone_details$total_dois</code></pre>
<pre><code>## [1] 236389</code></pre>
<p>Whether the data publishers provide on funders of articles they publish is current in Crossref (a TRUE/FALSE value–called “logical” in R):</p>
<pre class="r"><code># is funder data current on deposits?
plosone_details$deposits_funders_current</code></pre>
<pre><code>## [1] TRUE</code></pre>
<p><strong>TRY IT YOURSELF</strong></p>
<ol style="list-style-type: decimal">
<li>Assign the ISSN for <em>College &amp; Research Libraries</em> to a value - 2150-6701</li>
<li>Use the <code>query.author</code> field query (<code>flq</code>) to find all articles written by Lisa Janicke Hinchliffe</li>
</ol>
<pre class="r"><code># assign the C&amp;RL ISSN 2150-6701</code></pre>
<pre class="r"><code># use the query.author field query to search for articles written by Lisa
# Hinchliffe.</code></pre>
</div>
</div>
<div id="viewing-the-json-file" class="section level1">
<h1>Viewing the JSON file</h1>
<p>You can view these files in a JSON viewer using <code>jsonedit()</code> from the <code>listviewer</code> package. Note especially the last few variables. These are nested lists, as a single article can have multiple authors, and each author has a given name, family name, and sequence of authorship.</p>
<pre class="r"><code># assign the PLOS ISSN and get the last 25 articles on deposit
plosone_issn &lt;- &quot;1932-6203&quot;
plosone_publications &lt;- cr_journals(issn = plosone_issn, works = TRUE, limit = 5) %&gt;% 
    pluck(&quot;data&quot;)</code></pre>
<pre><code>## Warning: `tbl_df()` is deprecated as of dplyr 1.0.0.
## Please use `tibble::as_tibble()` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<pre class="r"><code>listviewer::jsonedit(plosone_publications, mode = &quot;view&quot;)</code></pre>
<div id="htmlwidget-fbb6d78e6e12754a8398" style="width:672px;height:480px;" class="jsonedit html-widget"></div>
<script type="application/json" data-for="htmlwidget-fbb6d78e6e12754a8398">{"x":{"data":{"container.title":["PLoS ONE","PLOS ONE","PLOS ONE","PLOS ONE","PLOS ONE"],"created":["2012-09-27","2015-03-20","2017-04-26","2015-05-15","2015-05-01"],"deposited":["2018-10-22","2019-08-22","2018-10-24","2018-10-23","2019-08-24"],"published.online":["2012-09-26","2015-03-20","2017-04-26","2015-05-15","2015-05-01"],"doi":["10.1371/journal.pone.0045138","10.1371/journal.pone.0115083","10.1371/journal.pone.0176491","10.1371/journal.pone.0126659","10.1371/journal.pone.0124629"],"indexed":["2020-03-30","2020-03-30","2020-03-30","2020-03-30","2020-03-30"],"issn":["1932-6203","1932-6203","1932-6203","1932-6203","1932-6203"],"issue":["9","3","4","5","5"],"issued":["2012-09-26","2015-03-20","2017-04-26","2015-05-15","2015-05-01"],"member":["340","340","340","340","340"],"page":["e45138","e0115083","e0176491","e0126659","e0124629"],"prefix":["10.1371","10.1371","10.1371","10.1371","10.1371"],"publisher":["Public Library of Science (PLoS)","Public Library of Science (PLoS)","Public Library of Science (PLoS)","Public Library of Science (PLoS)","Public Library of Science (PLoS)"],"reference.count":["0","50","38","18","27"],"score":["1","1","1","1","1"],"source":["Crossref","Crossref","Crossref","Crossref","Crossref"],"title":["Can Diversifying Selection Be Distinguished from History in Geographic Clines? A Population Genomic Study of Killifish (Fundulus heteroclitus)","Elevated Endogenous Erythropoietin Concentrations Are Associated with Increased Risk of Brain Damage in Extremely Preterm Neonates","Dominance of an alien shrub Rhus typhina over a native shrub Vitex negundo var. heterophylla under variable water supply patterns","Expenditure Analysis of HIV Testing and Counseling Services Using the Cascade Framework in Vietnam","Reed-Sternberg Cells Form by Abscission Failure in the Presence of Functional Aurora B Kinase"],"type":["journal-article","journal-article","journal-article","journal-article","journal-article"],"update.policy":["http://dx.doi.org/10.1371/journal.pone.corrections_policy","http://dx.doi.org/10.1371/journal.pone.corrections_policy","http://dx.doi.org/10.1371/journal.pone.corrections_policy","http://dx.doi.org/10.1371/journal.pone.corrections_policy","http://dx.doi.org/10.1371/journal.pone.corrections_policy"],"url":["http://dx.doi.org/10.1371/journal.pone.0045138","http://dx.doi.org/10.1371/journal.pone.0115083","http://dx.doi.org/10.1371/journal.pone.0176491","http://dx.doi.org/10.1371/journal.pone.0126659","http://dx.doi.org/10.1371/journal.pone.0124629"],"volume":["7","10","12","10","10"],"author":[{"given":["Allan E.","Larissa M.","Marjorie F.","Erik E."],"family":["Strand","Williams","Oleksiak","Sotka"],"sequence":["first","additional","additional","additional"]},{"given":["Steven J.","Elizabeth","J. Wells","Raina N.","Stephen","Karl C. K.","T. Michael","Nigel","Mari","Olaf","Alan",null],"family":["Korzeniewski","Allred","Logan","Fichorova","Engelke","Kuban","O’Shea","Paneth","Holm","Dammann","Leviton",null],"sequence":["first","additional","additional","additional","additional","additional","additional","additional","additional","additional","additional","additional"],"name":[null,null,null,null,null,null,null,null,null,null,null,"ELGAN study investigators"]},{"given":["Ning","Xiangfeng","Qiang","Xiao","Wenxin","Renqing","Jian","Weihua"],"family":["Du","Tan","Li","Liu","Zhang","Wang","Liu","Guo"],"sequence":["first","additional","additional","additional","additional","additional","additional","additional"]},{"given":["Van Thu","Huyen Thanh","Quoc Cuong","Phuong Thi Bich","Gary"],"family":["Nguyen","Nguyen","Nguyen","Duong","West"],"sequence":["first","additional","additional","additional","additional"]},{"given":["Ana","Helder","André F.","Susana A.","Patrícia","Wendy","William C.","Clara"],"family":["Xavier de Carvalho","Maiato","Maia","Ribeiro","Pontes","Bickmore","Earnshaw","Sambade"],"sequence":["first","additional","additional","additional","additional","additional","additional","additional"]}],"license":[{"date":["2012-09-26"],"URL":["http://creativecommons.org/licenses/by/4.0/"],"delay.in.days":[0],"content.version":["unspecified"]},{"date":["2015-03-20"],"URL":["http://creativecommons.org/licenses/by/4.0/"],"delay.in.days":[0],"content.version":["vor"]},{"date":["2017-04-26"],"URL":["http://creativecommons.org/licenses/by/4.0/"],"delay.in.days":[0],"content.version":["vor"]},{"date":["2015-05-15"],"URL":["http://creativecommons.org/licenses/by/4.0/"],"delay.in.days":[0],"content.version":["vor"]},{"date":["2015-05-01"],"URL":["http://creativecommons.org/licenses/by/4.0/"],"delay.in.days":[0],"content.version":["vor"]}],"link":[null,{"URL":["http://dx.plos.org/10.1371/journal.pone.0115083"],"content.type":["unspecified"],"content.version":["vor"],"intended.application":["similarity-checking"]},{"URL":["http://dx.plos.org/10.1371/journal.pone.0176491"],"content.type":["unspecified"],"content.version":["vor"],"intended.application":["similarity-checking"]},{"URL":["http://dx.plos.org/10.1371/journal.pone.0126659"],"content.type":["unspecified"],"content.version":["vor"],"intended.application":["similarity-checking"]},{"URL":["http://dx.plos.org/10.1371/journal.pone.0124629"],"content.type":["unspecified"],"content.version":["vor"],"intended.application":["similarity-checking"]}],"reference":[null,{"key":["ref1","ref2","ref3","ref4","ref5","ref6","ref7","ref8","ref9","ref10","ref11","ref12","ref13","ref14","ref15","ref16","ref17","ref18","ref19","ref20","ref21","ref22","ref23","ref24","ref25","ref26","ref27","ref28","ref29","ref30","ref31","ref32","ref33","ref34","ref35","ref36","ref37","ref38","ref39","ref40","ref41","ref42","ref43","ref44","ref45","ref46","ref47","ref48","ref49","ref50"],"doi.asserted.by":["crossref",null,"crossref","crossref",null,"crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref",null,"crossref",null,"crossref","crossref",null,"crossref","crossref","crossref","crossref","crossref",null,"crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref"],"first.page":["888","19488","105","e635","657","1023","E218","E681","325","897","395","104","362","64","315","22","719","45","566","68","66","5773","4741","e00280","418.e411","e00280","1201",null,"829",null,"103","395",null,"683","405","557","101","472","225","740","227","1032","1","142","187","269","387","971","128","604"],"DOI":["10.1016/0006-291X(92)91139-H",null,"10.1016/S0306-4522(96)00306-5","10.1542/peds.2005-3186",null,"10.1542/peds.2013-4307","10.1542/peds.2008-3553","10.1542/peds.2008-2701","10.1016/j.earlhumdev.2011.01.043","10.1016/j.jpeds.2010.11.059","10.1016/j.jpeds.2011.08.069","10.1016/j.bbi.2012.12.012","10.1038/pr.2012.188","10.1159/000084275","10.1097/SHK.0b013e3181fd0700","10.1016/j.cyto.2014.04.009","10.1016/j.earlhumdev.2009.08.060","10.1016/0378-3782(87)90099-5","10.1203/00006450-199911000-00013","10.1203/PDR.0b013e3181fed334","10.1016/j.cyto.2010.09.003","10.1128/IAI.00631-06","10.1021/ac702628q","10.1128/mBio.00280-10","10.1016/j.ajog.2010.12.010","10.1128/mBio.00280-10","10.1007/s00247-007-0605-z",null,"10.1177/08830738050200101001",null,"10.1016/j.ccr.2009.01.001","10.1016/j.jpeds.2011.08.069",null,"10.1542/peds.2012-0498","10.1111/j.1365-2796.2008.02024.x","10.1161/STROKEAHA.107.483008","10.1016/j.ydbio.2004.08.025","10.1203/00006450-200204000-00012",null,"10.1038/ki.1995.113","10.1016/j.expneurol.2011.04.021","10.1161/STROKEAHA.109.570325","10.1186/2040-7378-1-4","10.2169/internalmedicine.42.142","10.1111/j.1365-2249.2008.03760.x","10.1159/000105493","10.2174/156720205774962683","10.1084/jem.20021067","10.1016/S0006-8993(02)03239-0","10.1093/oxfordjournals.aje.a116133"],"article.title":["Production and ligand-binding characteristics of the soluble form of murine erythropoietin receptor","A novel site of erythropoietin production. Oxygen-dependent production in cultured rat astrocytes","Erythropoietin receptor is expressed in rat hippocampal and cerebral cortical neurons, and erythropoietin prevents in vitro glutamate-induced neuronal death","Erythropoietin concentrations and neurodevelopmental outcome in preterm infants","Erythropoietin Improves Neurodevelopmental Outcome of Extremely Preterm Infants","Cognitive Outcomes of Preterm Infants Randomized to Darbepoetin, Erythropoietin, or Placebo","Erythropoietin Improved Neurologic Outcomes in Newborns With Hypoxic-Ischemic Encephalopathy","Higher Cumulative Doses of Erythropoietin and Developmental Outcomes in Preterm Infants","Early postnatal blood concentrations of inflammation-related proteins and microcephaly two years later in infants born before the 28th post-menstrual week","The Relationship between Early Concentrations of 25 Blood Proteins and Cerebral White Matter Injury in Preterm Newborns: The ELGAN Study","Elevated Concentrations of Inflammation-Related Proteins in Postnatal Blood Predict Severe Developmental Delay at 2 Years of Age in Extremely Preterm Infants","Inflammation-initiating illnesses, inflammation-related proteins, and cognitive impairment in extremely preterm infants","Two-hit model of brain damage in the very preterm newborn: small for gestational age and postnatal systemic inflammation","Attenuation of interstitial inflammation and fibrosis by recombinant human erythropoietin in chronic cyclosporine nephropathy","Erythropoietin requires endothelial nitric oxide synthase to counteract TNF-[alpha]-induced microcirculatory dysfunction in murine striated muscle","Endogenous erythropoietin varies significantly with inflammation-related proteins in extremely premature newborns","The ELGAN study of the brain and related disorders in extremely low gestational age newborns","New birthweight and head circumference centiles for gestational ages 24 to 42 weeks","Maternal infection, fetal inflammatory response, and brain damage in very low birthweight infants","Relationship between neonatal blood protein profiles and placenta histologic characteristics in ELGANs","Inflammation-related proteins in the blood of extremely low gestational age newborns. The contribution of inflammation to the appearance of developmental regulation","Trichomonas vaginalis lipophosphoglycan triggers a selective upregulation of cytokines by human female reproductive tract epithelial cells","Biological and technical variables affecting immunoassay recovery of cytokines from human serum and simulated vaginal fluid: a multicenter study","Microbe-specific modulation of inflammatory response in extremely low gestational age newborns","Blood protein profiles of infants born before 28 weeks differ by pregnancy complication","Maternal microbe-specific modulation of inflammatory response in extremely low-gestational-age newborns","Observer variability assessing US scans of the preterm brain: the ELGAN study","Bayley Scales of Infant Development-II","Video and CD-ROM as a training tool for performing neurologic examinations of 1-year-old children in a multicenter epidemiologic study",null,"IL-6 and Stat3 are required for survival of intestinal epithelial cells and development of colitis-associated cancer","Elevated concentrations of inflammation-related proteins in postnatal blood predict severe developmental delay at 2 years of age in extremely preterm infants","Systemic Inflammation and Cerebral Palsy Risk in Extremely Preterm Infants","Erythropoietin for Neuroprotection in Neonatal Encephalopathy: Safety and Pharmacokinetics","Erythropoietin-mediated tissue protection: reducing collateral damage from the primary injury response","Erythropoietin enhances angiogenesis and neuronal replacement in cerebral cortex after neonatal hypoxia-ischemia","The role of erythropoietin in regulating angiogenesis","Erythropoietin stimulates vasculogenesis in neonatal rat mesenteric microvascular endothelial cells","Neuroprotection and angiogenesis: Dual role of erythropoietin in brain ischemia","RECOMBINANT-HUMAN-ERYTHROPOIETIN STIMULATES ANGIOGENESIS IN-VITRO","Neuroprotective potential of erythropoietin and its derivative carbamylated erythropoietin in periventricular leukomalacia","Enhanced Oligodendrogenesis and Recovery of Neurological Function by Erythropoietin After Neonatal Hypoxic/Ischemic Brain Injury","Erythropoietin: a multimodal neuroprotective agent","Pleiotropic functions of erythropoietin","Attenuation of monocyte proinflammatory cytokine responses to Neisseria meningitidis in children by erythropoietin","Erythropoietin attenuates lipopolysaccharide-induced white matter injury in the neonatal rat brain","Erythropoictin requires NF-kappa B and its nuclear translocation to prevent early and late apoptotic neuronal injury during beta-amyloid toxicity","Erythropoietin selectively attenuates cytokine production and inflammation in cerebral ischemia by targeting neuronal apoptosis","Erythropoietin exerts an anti-inflammatory effect on the CNS in a model of experimental autoimmune encephalomyelitis","Very low birth weight: a problematic cohort for epidemiologic studies of very small or immature neonates"],"volume":["188","269","76","118","67","133","124","124","87","158","160","29","73","25","35","69","85","15","46","69","53","74","80","2","204","2","37",null,"20",null,"15","160",null,"130","264","38","276","51","15","47","230","41","1","42","154","92","2","198","952","134"],"author":["M Nagao","S Masuda","E Morishita","R Bierer","A-P Neubauer","RK Ohls","C Zhu","MS Brown","A Leviton","A Leviton","TM O'Shea","TM O'Shea","A Leviton","SH Lee","C Contaldo","J Wells Logan","TM O'Shea","PL Yudkin","A Leviton","JL Hecht","A Leviton","RN Fichorova","RN Fichorova","RN Fichorova","TF McElrath","RN Fichorova","K Kuban","N Bayley","KC Kuban",null,"S Grivennikov","TM O'Shea","KC Kuban","YW Wu","M Brines","M Iwai","N Kertesz","RA Ashley","HH Marti","RG Carlini","W Liu","M Iwai","N Byts","R Sasaki","C Schultz","A Kumral","ZZ Chong","P Villa","D Agnello","CC Arnold"],"year":["1992","1994","1997","2006","2010","2014","2009","2009","2011","2011","2012","2013","2013","2005","2011","2014","2009","1987","1999","2011","2011","2006","2008","2011","2011","2011","2007","1993","2005",null,"2009","2012","2014","2012","2008","2007","2004","2002","2000","1995","2011","2010","2009","2003","2008","2007","2005","2003","2002","1991"],"journal.title":["Biochem Biophys Res Commun","J Biol Chem","Neuroscience","Pediatrics","Annals of Neurology","Pediatrics","Pediatrics","Pediatrics","Early Human Development","The Journal of Pediatrics","Journal of Pediatrics","Brain, Behavior, and Immunity","Pediatric Research","Am J Nephrol","Shock","Cytokine","Early Hum Dev","Early Hum Dev","Pediatric Research","Pediatr Research","Cytokine","Infect Immun","Anal Chem","mBio","Am J Obstet Gynecol","MBio","Pediatr Radiol",null,"J Child Neurol",null,"Cancer Cell","J Pediatr","J Child Neurol","Pediatrics","Journal of Internal Medicine","Stroke","Developmental Biology","Pediatric Research","News in Physiological Sciences","Kidney International","Experimental Neurology","Stroke","Experimental & Translational Stroke Medicine","Intern Med","Clinical and Experimental Immunology","Neonatology","Current Neurovascular Research","Journal of Experimental Medicine","Brain Research","Am J Epidemiol"],"unstructured":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"CDC (2007) CDC Growth Charts: United States <ext-link xmlns:xlink=\"http://www.w3.org/1999/xlink\" ext-link-type=\"uri\" xlink:href=\"http://www.cdc.gov/nchs/data/nhanes/growthcharts/zscore/zhcageinf.xls\" xlink:type=\"simple\">http://www.cdc.gov/nchs/data/nhanes/growthcharts/zscore/zhcageinf.xls<\/ext-link>.",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]},{"key":["ref1","ref2","ref3","ref4","ref5","ref6","ref7","ref8","ref9","ref10","ref11","ref12","ref13","ref14","ref15","ref16","ref17","ref18","ref19","ref20","ref21","ref22","ref23","ref24","ref25","ref26","ref27","ref28","ref29","ref30","ref31","ref32","ref33","ref34","ref35","ref36","ref37","ref38"],"doi.asserted.by":["publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher",null,null,"publisher","publisher","publisher","publisher","publisher",null,"publisher","publisher","publisher",null,"publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher","publisher"],"first.page":["319","1079","522","635","1853","345","93","389","63","501","1625","1079","353",null,null,"199","403","1307","1166","171",null,"e0140664","314","528","49","553","591","419","74","1147","1248","235","335","981","304","2202","130","293"],"DOI":["10.1111/j.1366-9516.2006.00249.x","10.1038/nature05719","10.1111/j.1744-7909.2008.00660.x","10.1111/j.1469-8137.2007.02189.x","10.1073/pnas.0808434106","10.1093/treephys/tpv123","10.1046/j.1472-4642.2000.00083.x","10.1007/s11104-013-1748-3","10.1007/s11104-014-2227-1","10.1890/06-0421","10.3732/ajb.1000125","10.1007/s11284-012-0989-6","10.1007/s00442-003-1435-9",null,null,"10.2307/3236604","10.1007/s10457-012-9562-0","10.1093/treephys/tpp065","10.1080/11263504.2013.852632","10.1016/j.jnc.2004.04.002",null,"10.1371/journal.pone.0140664","10.2307/1938948","10.1046/j.1365-2745.2000.00473.x",null,"10.1007/s11738-008-0265-0","10.1042/bst0110591","10.1111/j.1461-0248.2011.01596.x","10.1111/j.1399-3054.2012.01694.x","10.1111/j.1600-0706.2012.20877.x","10.1111/j.1365-2435.2011.01886.x","10.1111/j.1461-0248.2009.01418.x","10.1071/BT02124","10.1111/j.1461-0248.2006.00950.x","10.2307/2389785","10.1126/science.1076347","10.1890/08-2039.1","10.1023/B:BINV.0000005574.15074.66"],"article.title":["Predicting invasions by woody species in a temperate zone: a test of three risk assessment schemes in the Czech Republic (Central Europe)","Resource-use efficiency and plant invasion in low-resource systems","Invasion possibility and potential effects of <italic>Rhus typhina<\/italic> on Beijing municipality","Leaf traits relationships of native and invasive plants: community- and global-scale comparisons","Evolutionary tradeoffs for nitrogen allocation to photosynthesis versus cell walls in an invasive plant","Functional traits contributed to the superior performance of the exotic species <italic>Robinia pseudoacacia<\/italic>: a comparison with the native tree <italic>Sophora japonica<\/italic>","Naturalization and invasion of alien species: concepts and definitions","Competitive interaction between the exotic plant <italic>Rhus typhina<\/italic> L. and the native tree <italic>Quercus acutissima<\/italic> Carr. in Northern China under different soil N:P ratios","Increased nitrogen deposition alleviated the competitive effects on the introduced invasive plant <italic>Robinia pseudoacacia<\/italic> on the native tree <italic>Quercus acutissima<\/italic>","Amount or pattern? Grassland responses to the heterogeneity and availability of two key resources","The impact of altered precipitation variability on annual weed species","Effects of temporal heterogeneity of water supply and nutrient levels on plant biomass growth depend on the plant’s relative size within its population","The effects of mean and variance in resource supply on survival of annuals from Mediterranean and desert environments",null,null,"Effects of water pulsing on individual performance and competitive hierarchies in plants","Morphological responses of <italic>Vitex negundo<\/italic> var. <italic>heterophylla<\/italic> and <italic>Ziziphus jujuba<\/italic> var. <italic>spinosa<\/italic> to the combined impact of drought and shade","Ecophysiological evaluation of the potential invasiveness of <italic>Rhus typhina<\/italic> in its non-native habitats","Challenge of weed risk assessment (WRA) for ecological restoration in China: The case of <italic>Rhus typhina<\/italic> L. and the officially released weed risk assessment system","Assessing the risk of potentially invasive plant species in central Europe",null,"Assessing the Capacity of Plant Species to Accumulate Particulate Matter in Beijing, China","Competitive interactions in plant communities of different successional ages","Fluctuating resources in plant communities: a general theory of invisibility","Ecophysiology of photosynthesis","Leaf movement and photosynthetic plasticity of black locust (<italic>Robinia pseudoacacia<\/italic>) alleviate stress under different light and water conditions","Determination of total carotenoids and chlorophylls a and b of leaf extracts in different solvents","Do invasive species show higher phenotypic plasticity than native species and, if so, is it adaptive? A meta-analysis","Is hemiepiphytism an adaptation to high irradiance? Testing seedling responses to light levels and drought in hemiepiphytic and non-hemiepiphytic <italic>Ficus<\/italic>","The interplay between above- and below-ground plant-plant interactions along an environmental gradient: insights from two-layer zone-of-influence models","Multispecies comparison reveals that invasive and native plants differ in their traits but not in their plasticity","A meta-analysis of trait differences between invasive and non-invasive plant species","A handbook of protocols for standardized and easy measurement of plant functional traits worldwide","Jack of all trades, master of some? On the role of phenotypic plasticity in plant invasions","On the meaning of competition and the mechanisms of competitive superiority","Rainfall variability, carbon cycling, and plant species diversity in a mesic grassland","Do competitors modulate rare plant response to precipitation change?","Human Agency in biological invasions: Secondary releases foster naturalisation and population expansion of exotic plant species"],"volume":["12","446","50","176","106","36","6","372","385","88","97","27","141","4",null,"12","87","29","147","12",null,"10","63","88",null,"31","11","14","148","122","25","13","51","9","1","298","91","5"],"author":["M Křivánek","J Funk","GM Wang","MR Leishman","YL Feng","YJ Luo","DM Richardson","YF Yuan","YJ Luo","FT Maestre","TMP Robinson","Y Hagiwara","AA Sher","S Solomon",null,"A Novoplansky","N Du","ZJ Zhang","GM Wang","E Weber",null,"L Mo","JAD Parrish","MA Davis","U Schreiber","F Xu","HK Lichtenthaler","AM Davidson","GY Hao","WP Zhang","O Godoy","M Van Kleunen","JHC Cornelissen","CL Richards","D Tilman","AK Knapp","JM Levine","I Kowarik"],"year":["2006","2007","2008","2007","2009","2016","2000","2013","2014","2007","2010","2012","2004","2007",null,"2001","2013","2009","2013","2004",null,"2015","1982","2000","1994","2009","1983","2011","2013","2013","2011","2010","2003","2006","1987","2002","2010","2003"],"journal.title":["Divers Distrib","Nature","J Integr Plant Biol","New Phytol","Proc Natl Acad Sci USA","Tree Physiol","Divers Distrib","Plant Soil","Plant Soil","Ecology","Am J Bot","Ecological research","Oecologia",null,null,"J Veg Sci","Agroforest Syst","Tree Physiol","Plant Biosyst","J Nat Conserv",null,"PLoS ONE","Ecology","J Ecol",null,"Acta Physiol Plant","Biochem Soc Trans","Ecol Lett","Physiol Plant","Oikos","Funct Ecol","Ecol Lett","Aust J Bot","Ecol Lett","Funct Ecol","Science","Ecology","Biol Invasions"],"issue":[null,null,null,null,null,null,null,null,null,"2","10",null,null,null,null,null,null,null,null,null,null,"10","2",null,null,null,null,null,null,null,null,null,null,null,null,null,"1",null],"unstructured":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Hijioka Y, Lin E, Pereira JJ, Corlett RT, Cui X, Insarov GE, et al. Asia. In: Barros VR, Field CB, Dokken DJ, Mastrandrea MD, Mach KJ, Bilir TE, et al. Climate Change 2014: Impacts, Adaptation, and Vulnerability. Part B: Regional Aspects. Contribution of Working Group II to the Fifth Assessment Report of the Intergovernmental Panel on Climate Change. Cambridge University Press, Cambridge, United Kingdom and New York, NY, USA; 2014. pp. 1327–1370.",null,null,null,null,null,"Wittenberg R (ed.). An inventory of alien species and their threat to biodiversity and economy in Switzerland. CABI Bioscience Switzerland Centre report to the Swiss Agency for Environment, Forests and Landscape; 2005. pp. 401–402.",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]},{"key":["ref1","ref2","ref3","ref4","ref5","ref6","ref7","ref8","ref9","ref10","ref11","ref12","ref13","ref14","ref15","ref16","ref17","ref18"],"year":["2012","2004","2014","2009","2005","2012","2014","2013","2012","2012","2011","2014","2014","2001",null,"2014","2013","1996"],"article.title":["Vietnam AIDS Response Progress report","HIV/AIDS epidemics in Vietnam: evolution and responses","Summary of activities response to HIV/AIDS epidemic in Vietnam 2013 and plan for 2014","Financing the Response to HIV in Low-Income and Middle-Income Countries","Cost effectiveness analysis of strategies to combat HIV/AIDS in developing countries","Vietnam National AIDS Spending Assessment 2008–2010","Vietnam National AIDS Spending Assessment 2011–2012","Cost effectiveness of facility and home based HIV voluntary counseling and testing strategies in rural Uganda","Optimising the cost and delivery of HIV counselling and testing services in Kenya and Swaziland","The Cost of Providing HIV/AIDS Counseling and Testing Services in Vietnam","The Spectrum of Engagement in HIV Care and its Relevance to Test-and-Treat Strategies for Prevention of HIV Infection","The HIV care cascade through time","Report on HIV/AIDS prevention and control activities in 2013","Economic evaluation in health care: merging theory with practice",null,"Is there scope for cost savings and efficiency gains in HIV services? A systematic review of the evidence from low-and middle-income countries","Vietnam HIV/AIDS Estimates and Projections 2015–2020","Recommendations for reporting cost-effectiveness analyses"],"issue":[null,"3 Suppl A",null,null,"7530",null,null,"2","7","1","6","1",null,null,null,"7",null,"16"],"first.page":[null,"137",null,"S119","1431",null,null,"423","498","36","793","5",null,null,null,"499",null,"1339"],"volume":[null,"16",null,"52","331",null,null,"13","88","1","52","14",null,null,null,"92",null,"276"],"author":[null,"TH Nguyen",null,"J Antonio Izazola-Licea","DR Hogan",null,null,"EM Mulogo","CD Obure","HV Minh","EM Gardner","EM Gardner",null,"MF Drummond",null,"M Siapka",null,"JE Siegel"],"journal.title":[null,"AIDS Educ Prev",null,"JAIDS Journal of Acquired Immune Deficiency Syndromes","BMJ",null,null,"Afr Health Sci","Sex Transm Infect","Value in Health Regional Issues","Clinical Infectious Diseases","Lancet Infect Dis",null,null,null,"Bulletin of the World Health Organization",null,"JAMA"],"doi.asserted.by":[null,null,null,"crossref","crossref",null,null,null,"crossref","crossref","crossref","crossref",null,null,null,"crossref",null,"crossref"],"DOI":[null,null,null,"10.1097/QAI.0b013e3181baeeda","10.1136/bmj.38643.368692.68",null,null,null,"10.1136/sextrans-2012-050544","10.1016/j.vhri.2012.03.012","10.1093/cid/ciq243","10.1016/S1473-3099(13)70272-X",null,null,null,"10.2471/BLT.13.127639",null,"10.1001/jama.1996.03540160061034"],"unstructured":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,"The World Bank. Official exchange rate [cited 3 October 2014]. In: The World Bank Group [Internet]; 2014. Available: <ext-link xmlns:xlink=\"http://www.w3.org/1999/xlink\" ext-link-type=\"uri\" xlink:href=\"http://data.worldbank.org/indicator/PA.NUS.FCRF\" xlink:type=\"simple\">http://data.worldbank.org/indicator/PA.NUS.FCRF<\/ext-link>",null,null,null]},{"issue":["23","7","4","1","12","2","4","5","6","6","51","6","5","3","8","4","3","1","12","9",null,"3","2","13","1","35","46"],"key":["ref1","ref2","ref3","ref4","ref5","ref6","ref7","ref8","ref9","ref10","ref11","ref12","ref13","ref14","ref15","ref16","ref17","ref18","ref19","ref20","ref21","ref22","ref23","ref24","ref25","ref26","ref27"],"doi.asserted.by":["crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref",null,"crossref","crossref","crossref",null,"crossref",null,"crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref","crossref"],"first.page":["10962","453","1495","15","4283","197","769","1092","417","1083","20729","3081","1382","289","3429","529","818","6","789","1106",null,"473","179","12997","99","14920","39083"],"DOI":["10.1073/pnas.91.23.10962","10.1056/NEJM199708143370703","10.1084/jem.184.4.1495","10.1038/nrc2542","10.1182/blood-2002-01-0099","10.1002/ijc.11370","10.1002/ijc.23119","10.1002/ijc.24389",null,"10.1002/ijc.2910430622","10.1073/pnas.1312509110","10.2353/ajpath.2010.100089",null,"10.1038/labinvest.3780237",null,"10.1172/JCI200316624","10.1182/blood.V97.3.818","10.1002/ajh.1068","10.1038/nrm3474","10.1016/j.bcp.2011.02.010","10.1007/s10735-013-9561-0","10.1016/j.cell.2008.12.020","10.1083/jcb.200404001","10.1074/jbc.M311128200","10.1016/S0092-8674(00)80718-7","10.1073/pnas.0904231106","10.1074/jbc.M112.390088"],"article.title":["Hodgkin disease: Hodgkin and Reed-Sternberg cells picked from histological sections show clonal immunoglobulin gene rearrangements and appear to be derived from B cells at various stages of development","Origin of nodular lymphocyte-predominant Hodgkin&apos;s disease from a clonal expansion of highly mutated germinal-center B cells","Hodgkin and Reed-Sternberg cells in Hodgkin&apos;s disease represent the outgrowth of a dominant tumor clone derived from (crippled) germinal center B cells","The biology of Hodgkin&apos;s lymphoma","The role of cytokines in classical Hodgkin lymphoma","Expression of CCL5/RANTES by Hodgkin and Reed-Sternberg cells and its possible role in the recruitment of mast cells into lymphomatous tissue","Expression of CCR5 receptors on Reed-Sternberg cells and Hodgkin lymphoma cell lines: involvement of CCL5/Rantes in tumor cell growth and microenvironmental interactions","Functional coexpression of Interleukin (IL)-7 and its receptor (IL-7R) on Hodgkin and Reed-Sternberg cells: Involvement of IL-7 in tumor cell growth and microenvironmental interactions of Hodgkin&apos;s lymphoma","L-428 Reed-Sternberg cells and mononuclear Hodgkin&apos;s cells arise from a single cloned mononuclear cell","Formation of multinucleated cells in a Hodgkin&apos;s-disease-derived cell line","Incomplete cytokinesis and re-fusion of small mononucleated Hodgkin cells lead to giant multinucleated Reed–Sternberg cells","Tumorigenic Potential of Mononucleated Small Cells of Hodgkin Lymphoma Cell Lines","Reed-Sternberg cells in Hodgkin&apos;s cell lines HDLM, L-428, and KM-H2 are not actively replicating: lack of bromodeoxyuridine uptake by multinuclear cells in culture","Molecular single-cell analysis of Hodgkin- and Reed-Sternberg cells harboring unmutated immunoglobulin variable region genes","Molecular single cell analysis demonstrates the derivation of a peripheral blood-derived cell line (L1236) from the Hodgkin/Reed-Sternberg cells of a Hodgkin&apos;s lymphoma patient","Identification of Hodgkin and Reed-Sternberg cell-specific genes by gene expression profiling","Evidence that Hodgkin and Reed-Sternberg cells in Hodgkin disease do not represent cell fusions","Cell fusion is not involved in the generation of giant cells in the Hodgkin-Reed Sternberg cell line L1236","The chromosomal passengercomplex (CPC): from easy rider to thegodfather of mitosis","Effects of AZD1152, a selective Aurora B kinase inhibitor, on Burkitt’s and Hodgkin’s lymphomas",null,"Aurora B-Mediated Abscission Checkpoint Protects against Tetraploidization","Borealin: a novel chromosomal passenger required for stability of the bipolar mitotic spindle","Autophosphorylation of a Newly Identified Site of Aurora-B Is Indispensable for Cytokinesis","Phosphorylation of histone H3 is required for proper chromosome condensation and segregation","Mutations in a gene encoding a midbody kelch protein in familial and sporadic classical Hodgkin lymphoma lead to binucleated cells","The Kelch Protein KLHDC8B Guards against Mitotic Errors, Centrosomal Amplification, and Chromosomal Instability"],"volume":["91","337","184","9","99","107","122","125","6","43","110","177","71","81","87","111","97","67","13","81",null,"136","166","279","97","106","287"],"author":["R Küppers","T Marafioti","H Kanzler","R Küppers","BF Skinnider","M Fischer","D Aldinucci","L Cattaruzza","SR Newcom","HG Drexler","B Rengstl","J-I Ikeda","SM Hsu","M Müschen","H Kanzler","R Küppers","R Küppers","D Re","M Carmena","N Mori",null,"P Steigemann","R Gassmann","Y Yasui","Y Wei","SJ Salipante","MM Krem"],"year":["1994","1997","1996","2008","2002","2003","2008","2009","1988","1989","2013","2010","1988","2001","1996","2003","2001","2001","2012","2011",null,"2009","2004","2004","1999","2009","2012"],"journal.title":["Proc Natl Acad Sci USA","N. Engl. J. Med.","J. Exp. Med.","Nat Rev Cancer","Blood","Int J Cancer","Int J Cancer","Int J Cancer","Int. J. Cell.","Int J Cancer","Proc Natl Acad Sci USA","The American Journal of Pathology","Blood","Lab Invest","Blood","J. Clin. Invest.","Blood","Am J Hematol","Nat Rev Mol Cell Biol","Biochemical Pharmacology",null,"Cell","J Cell Biol","Journal of Biological Chemistry","Cell","Proc Natl Acad Sci USA","Journal of Biological Chemistry"],"unstructured":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Ioannou M, Kouvaras E, Stathakis E, Samara M, Koukoulis GK. Aurora B kinase in Hodgkin lymphoma: immunohistochemical pattern of expression in neoplastic Hodgkin and Reed-Sternberg cells. J Mol Hist. 2013 Dec 24.",null,null,null,null,null,null]}],"funder":[null,null,{"DOI":["10.13039/501100001809","10.13039/501100007129",null],"name":["National Natural Science Foundation of China","Natural Science Foundation of Shandong Province","the Fundamental Research Funds of Shandong University"],"doi.asserted.by":["publisher","publisher",null],"award1":["31470402",null,null],"award2":["31400173",null,null],"award":[null,"ZR2014CQ038","2014GN023"]},null,null]},"options":{"mode":"view","modes":["code","form","text","tree","view"]}},"evals":[],"jsHooks":[]}</script>
<p>To save the JSON output, first use <code>toJSON()</code>, then <code>write_json()</code> from <code>jsonlite</code>.</p>
<pre class="r"><code># use the toJSON function to convert the output to JSON
plosone_publications_json &lt;- jsonlite::toJSON(plosone_publications)

# write a JSON file
jsonlite::write_json(plosone_publications_json, &quot;plosone_publications.json&quot;)</code></pre>
<p>You can also open the json in an online json viewer such as <a href="https://codebeautify.org/jsonviewer">Code Beautify</a>. Either open a file or paste the JSON output on the left side. Click <strong>Tree Viewer</strong> to view the data.</p>
</div>
<div id="using-cr_works-to-get-data-on-articles" class="section level1">
<h1>Using <code>cr_works()</code> to get data on articles</h1>
<p><code>cr_works()</code> allows you to search by DOI or a general query in order to return the Crossref metadata.</p>
<p>It is important to note, as Crossref does <a href="https://github.com/CrossRef/rest-api-doc/blob/master/demos/crossref-api-demo.ipynb">in the documentation</a>:</p>
<blockquote>
<p>Crossref does not use “works” in the FRBR sense of the word. In Crossref parlance, a “work” is just a thing identified by a DOI. In practice, Crossref DOIs are used as citation identifiers. So, in FRBR terms, this means, that a Crossref DOI tends to refer to one <em>expression</em> which might include multiple <em>manifestations</em>. So, for example, the ePub, HTML and PDF version of an article will share a Crossref DOI because the differences between them should not effect the interpretation or crediting of the content. In short, they can be cited interchangeably. The same is true of the “accepted manuscript” and the “version-of-record” of that accepted manuscript.</p>
</blockquote>
<div id="searching-by-doi" class="section level2">
<h2>Searching by DOI</h2>
<p>You can pass a DOI directly to <code>cr_works()</code> using the <code>dois</code> argument:</p>
<pre class="r"><code># Get metadata for a single article by DOI
jlsc_ku_oa &lt;- cr_works(dois = &quot;10.7710/2162-3309.1252&quot;) %&gt;% purrr::pluck(&quot;data&quot;)

# print the data frame with select columns
jlsc_ku_oa %&gt;% dplyr::select(title, doi, volume, issue, page, issued, url, publisher, 
    reference.count, type, issn)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["title"],"name":[1],"type":["chr"],"align":["left"]},{"label":["doi"],"name":[2],"type":["chr"],"align":["left"]},{"label":["volume"],"name":[3],"type":["chr"],"align":["left"]},{"label":["issue"],"name":[4],"type":["chr"],"align":["left"]},{"label":["page"],"name":[5],"type":["chr"],"align":["left"]},{"label":["issued"],"name":[6],"type":["chr"],"align":["left"]},{"label":["url"],"name":[7],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[8],"type":["chr"],"align":["left"]},{"label":["reference.count"],"name":[9],"type":["chr"],"align":["left"]},{"label":["type"],"name":[10],"type":["chr"],"align":["left"]},{"label":["issn"],"name":[11],"type":["chr"],"align":["left"]}],"data":[{"1":"Campus Open Access Funds: Experiences of the KU “One University” Open\\n                        Access Author Fund","2":"10.7710/2162-3309.1252","3":"3","4":"1","5":"1252","6":"2015-09-18","7":"http://dx.doi.org/10.7710/2162-3309.1252","8":"Pacific University Library","9":"0","10":"journal-article","11":"2162-3309"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>You can also pass more than one DOI. Here we start by assigning our DOIs to a variable <code>my_dois</code>, then pass it to <code>cr_works()</code> in the <code>doi</code> argument:</p>
<pre class="r"><code># Use c() to create a vector of DOIs
my_dois &lt;- c(&quot;10.2139/ssrn.2697412&quot;, &quot;10.1016/j.joi.2016.08.002&quot;, &quot;10.1371/journal.pone.0020961&quot;, 
    &quot;10.3389/fpsyg.2018.01487&quot;, &quot;10.1038/d41586-018-00104-7&quot;, &quot;10.12688/f1000research.8460.2&quot;, 
    &quot;10.7551/mitpress/9286.001.0001&quot;)

# pass the my_dois vector to cr_works()
my_dois_works &lt;- rcrossref::cr_works(dois = my_dois) %&gt;% pluck(&quot;data&quot;)

# print the data frame with select columns
my_dois_works %&gt;% dplyr::select(title, doi, volume, issue, page, issued, url, publisher, 
    reference.count, type, issn)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["title"],"name":[1],"type":["chr"],"align":["left"]},{"label":["doi"],"name":[2],"type":["chr"],"align":["left"]},{"label":["volume"],"name":[3],"type":["chr"],"align":["left"]},{"label":["issue"],"name":[4],"type":["chr"],"align":["left"]},{"label":["page"],"name":[5],"type":["chr"],"align":["left"]},{"label":["issued"],"name":[6],"type":["chr"],"align":["left"]},{"label":["url"],"name":[7],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[8],"type":["chr"],"align":["left"]},{"label":["reference.count"],"name":[9],"type":["chr"],"align":["left"]},{"label":["type"],"name":[10],"type":["chr"],"align":["left"]},{"label":["issn"],"name":[11],"type":["chr"],"align":["left"]}],"data":[{"1":"Open Access Publishing: A Literature Review","2":"10.2139/ssrn.2697412","3":"NA","4":"NA","5":"NA","6":"2014","7":"http://dx.doi.org/10.2139/ssrn.2697412","8":"Elsevier BV","9":"0","10":"journal-article","11":"1556-5068"},{"1":"Hybrid open access—A longitudinal study","2":"10.1016/j.joi.2016.08.002","3":"10","4":"4","5":"919-932","6":"2016-11","7":"http://dx.doi.org/10.1016/j.joi.2016.08.002","8":"Elsevier BV","9":"33","10":"journal-article","11":"1751-1577"},{"1":"The Development of Open Access Journal Publishing from 1993 to 2009","2":"10.1371/journal.pone.0020961","3":"6","4":"6","5":"e20961","6":"2011-06-13","7":"http://dx.doi.org/10.1371/journal.pone.0020961","8":"Public Library of Science (PLoS)","9":"31","10":"journal-article","11":"1932-6203"},{"1":"The Impact Factor Fallacy","2":"10.3389/fpsyg.2018.01487","3":"9","4":"NA","5":"NA","6":"2018-08-20","7":"http://dx.doi.org/10.3389/fpsyg.2018.01487","8":"Frontiers Media SA","9":"52","10":"journal-article","11":"1664-1078"},{"1":"Funders should mandate open citations","2":"10.1038/d41586-018-00104-7","3":"553","4":"7687","5":"129-129","6":"2018-01","7":"http://dx.doi.org/10.1038/d41586-018-00104-7","8":"Springer Science and Business Media LLC","9":"0","10":"journal-article","11":"0028-0836,1476-4687"},{"1":"The academic, economic and societal impacts of Open Access: an evidence-based review","2":"10.12688/f1000research.8460.2","3":"5","4":"NA","5":"632","6":"2016-06-09","7":"http://dx.doi.org/10.12688/f1000research.8460.2","8":"F1000 Research Ltd","9":"184","10":"journal-article","11":"2046-1402"},{"1":"Open Access","2":"10.7551/mitpress/9286.001.0001","3":"NA","4":"NA","5":"NA","6":"2012","7":"http://dx.doi.org/10.7551/mitpress/9286.001.0001","8":"The MIT Press","9":"0","10":"book","11":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="unnesting-list-columns" class="section level2">
<h2>Unnesting list columns</h2>
<p>Authors, links, licenses, funders, and some other values can appear in nested lists when you call <code>cr_journals</code> because there can be, and often are, multiple of each of these items per article. You can check the data classes on all variables by running <code>typeof()</code> across all columns using the <code>map_chr()</code> function from <code>purrr</code>:</p>
<pre class="r"><code># query to get data on a specific PLOS article
plos_article &lt;- cr_works(dois = &quot;10.1371/journal.pone.0228782&quot;) %&gt;% purrr::pluck(&quot;data&quot;)

# print the type of each column (e.g. character, numeric, logical, list)
purrr::map_chr(plos_article, typeof)</code></pre>
<pre><code>##  container.title          created        deposited published.online 
##      &quot;character&quot;      &quot;character&quot;      &quot;character&quot;      &quot;character&quot; 
##              doi          indexed             issn            issue 
##      &quot;character&quot;      &quot;character&quot;      &quot;character&quot;      &quot;character&quot; 
##           issued           member             page           prefix 
##      &quot;character&quot;      &quot;character&quot;      &quot;character&quot;      &quot;character&quot; 
##        publisher  reference.count            score           source 
##      &quot;character&quot;      &quot;character&quot;      &quot;character&quot;      &quot;character&quot; 
##          subject            title             type    update.policy 
##      &quot;character&quot;      &quot;character&quot;      &quot;character&quot;      &quot;character&quot; 
##              url           volume           author           funder 
##      &quot;character&quot;      &quot;character&quot;           &quot;list&quot;           &quot;list&quot; 
##             link          license        reference 
##           &quot;list&quot;           &quot;list&quot;           &quot;list&quot;</code></pre>
<p>Our <code>plos_article</code> data frame has a nested list for <strong>author</strong>. We can unnest this column using <code>unnest()</code> from the <code>tidyr</code> package. The <code>.drop = TRUE</code> argument will drop any other list columns.</p>
<pre class="r"><code># unnest author column
plos_article %&gt;% tidyr::unnest(author, .drop = TRUE)</code></pre>
<pre><code>## Warning: The `.drop` argument of `unnest()` is deprecated as of tidyr 1.0.0.
## All list-columns are now preserved.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["container.title"],"name":[1],"type":["chr"],"align":["left"]},{"label":["created"],"name":[2],"type":["chr"],"align":["left"]},{"label":["deposited"],"name":[3],"type":["chr"],"align":["left"]},{"label":["published.online"],"name":[4],"type":["chr"],"align":["left"]},{"label":["doi"],"name":[5],"type":["chr"],"align":["left"]},{"label":["indexed"],"name":[6],"type":["chr"],"align":["left"]},{"label":["issn"],"name":[7],"type":["chr"],"align":["left"]},{"label":["issue"],"name":[8],"type":["chr"],"align":["left"]},{"label":["issued"],"name":[9],"type":["chr"],"align":["left"]},{"label":["member"],"name":[10],"type":["chr"],"align":["left"]},{"label":["page"],"name":[11],"type":["chr"],"align":["left"]},{"label":["prefix"],"name":[12],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[13],"type":["chr"],"align":["left"]},{"label":["reference.count"],"name":[14],"type":["chr"],"align":["left"]},{"label":["score"],"name":[15],"type":["chr"],"align":["left"]},{"label":["source"],"name":[16],"type":["chr"],"align":["left"]},{"label":["subject"],"name":[17],"type":["chr"],"align":["left"]},{"label":["title"],"name":[18],"type":["chr"],"align":["left"]},{"label":["type"],"name":[19],"type":["chr"],"align":["left"]},{"label":["update.policy"],"name":[20],"type":["chr"],"align":["left"]},{"label":["url"],"name":[21],"type":["chr"],"align":["left"]},{"label":["volume"],"name":[22],"type":["chr"],"align":["left"]},{"label":["ORCID"],"name":[23],"type":["chr"],"align":["left"]},{"label":["authenticated.orcid"],"name":[24],"type":["lgl"],"align":["right"]},{"label":["given"],"name":[25],"type":["chr"],"align":["left"]},{"label":["family"],"name":[26],"type":["chr"],"align":["left"]},{"label":["sequence"],"name":[27],"type":["chr"],"align":["left"]},{"label":["funder"],"name":[28],"type":["list"],"align":["right"]},{"label":["link"],"name":[29],"type":["list"],"align":["right"]},{"label":["license"],"name":[30],"type":["list"],"align":["right"]},{"label":["reference"],"name":[31],"type":["list"],"align":["right"]}],"data":[{"1":"PLOS ONE","2":"2020-03-04","3":"2020-03-04","4":"2020-03-04","5":"10.1371/journal.pone.0228782","6":"2020-06-10","7":"1932-6203","8":"3","9":"2020-03-04","10":"340","11":"e0228782","12":"10.1371","13":"Public Library of Science (PLoS)","14":"73","15":"1","16":"Crossref","17":"General Biochemistry, Genetics and Molecular Biology,General Agricultural and Biological Sciences,General Medicine","18":"Learning about the Ellsberg Paradox reduces, but does not abolish, ambiguity aversion","19":"journal-article","20":"http://dx.doi.org/10.1371/journal.pone.corrections_policy","21":"http://dx.doi.org/10.1371/journal.pone.0228782","22":"15","23":"http://orcid.org/0000-0001-5870-9336","24":"TRUE","25":"Ruonan","26":"Jia","27":"first","28":"<tibble>","29":"<tibble>","30":"<tibble>","31":"<tibble>"},{"1":"PLOS ONE","2":"2020-03-04","3":"2020-03-04","4":"2020-03-04","5":"10.1371/journal.pone.0228782","6":"2020-06-10","7":"1932-6203","8":"3","9":"2020-03-04","10":"340","11":"e0228782","12":"10.1371","13":"Public Library of Science (PLoS)","14":"73","15":"1","16":"Crossref","17":"General Biochemistry, Genetics and Molecular Biology,General Agricultural and Biological Sciences,General Medicine","18":"Learning about the Ellsberg Paradox reduces, but does not abolish, ambiguity aversion","19":"journal-article","20":"http://dx.doi.org/10.1371/journal.pone.corrections_policy","21":"http://dx.doi.org/10.1371/journal.pone.0228782","22":"15","23":"NA","24":"NA","25":"Ellen","26":"Furlong","27":"additional","28":"<tibble>","29":"<tibble>","30":"<tibble>","31":"<tibble>"},{"1":"PLOS ONE","2":"2020-03-04","3":"2020-03-04","4":"2020-03-04","5":"10.1371/journal.pone.0228782","6":"2020-06-10","7":"1932-6203","8":"3","9":"2020-03-04","10":"340","11":"e0228782","12":"10.1371","13":"Public Library of Science (PLoS)","14":"73","15":"1","16":"Crossref","17":"General Biochemistry, Genetics and Molecular Biology,General Agricultural and Biological Sciences,General Medicine","18":"Learning about the Ellsberg Paradox reduces, but does not abolish, ambiguity aversion","19":"journal-article","20":"http://dx.doi.org/10.1371/journal.pone.corrections_policy","21":"http://dx.doi.org/10.1371/journal.pone.0228782","22":"15","23":"NA","24":"NA","25":"Sean","26":"Gao","27":"additional","28":"<tibble>","29":"<tibble>","30":"<tibble>","31":"<tibble>"},{"1":"PLOS ONE","2":"2020-03-04","3":"2020-03-04","4":"2020-03-04","5":"10.1371/journal.pone.0228782","6":"2020-06-10","7":"1932-6203","8":"3","9":"2020-03-04","10":"340","11":"e0228782","12":"10.1371","13":"Public Library of Science (PLoS)","14":"73","15":"1","16":"Crossref","17":"General Biochemistry, Genetics and Molecular Biology,General Agricultural and Biological Sciences,General Medicine","18":"Learning about the Ellsberg Paradox reduces, but does not abolish, ambiguity aversion","19":"journal-article","20":"http://dx.doi.org/10.1371/journal.pone.corrections_policy","21":"http://dx.doi.org/10.1371/journal.pone.0228782","22":"15","23":"NA","24":"NA","25":"Laurie R.","26":"Santos","27":"additional","28":"<tibble>","29":"<tibble>","30":"<tibble>","31":"<tibble>"},{"1":"PLOS ONE","2":"2020-03-04","3":"2020-03-04","4":"2020-03-04","5":"10.1371/journal.pone.0228782","6":"2020-06-10","7":"1932-6203","8":"3","9":"2020-03-04","10":"340","11":"e0228782","12":"10.1371","13":"Public Library of Science (PLoS)","14":"73","15":"1","16":"Crossref","17":"General Biochemistry, Genetics and Molecular Biology,General Agricultural and Biological Sciences,General Medicine","18":"Learning about the Ellsberg Paradox reduces, but does not abolish, ambiguity aversion","19":"journal-article","20":"http://dx.doi.org/10.1371/journal.pone.corrections_policy","21":"http://dx.doi.org/10.1371/journal.pone.0228782","22":"15","23":"NA","24":"NA","25":"Ifat","26":"Levy","27":"additional","28":"<tibble>","29":"<tibble>","30":"<tibble>","31":"<tibble>"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>We can see this has added 5 rows and 3 new columns: <strong>given</strong> (first name), <strong>family</strong> (last name), and <strong>sequence</strong> (order in which they appeared).</p>
<p>See <a href="https://ciakovx.github.io/rcrossref.html#unnesting_list_columns" class="uri">https://ciakovx.github.io/rcrossref.html#unnesting_list_columns</a> for more detailed strategies in unnesting nested lists in Crossref. For more details, call <code>?unnest</code> and read the <a href="https://r4ds.had.co.nz/many-models.html#unnesting">R for Data Science section on Unnesting</a>.</p>
<hr />
<p><strong>TRY IT YOURSELF</strong></p>
<ol style="list-style-type: decimal">
<li>Refer back to the type of each column above. Try unnesting one of the other list columns. Scroll all the way to the right. What new columns have appeared? Have any rows been duplicated?</li>
</ol>
</div>
<div id="getting-more-than-1000-results-with-the-cursor-argument-to-cr_journals" class="section level2">
<h2>Getting more than 1000 results with the <code>cursor</code> argument to <code>cr_journals</code></h2>
<p>If our result will have more than 1000 results, we have to use the <code>cursor</code> argument. Here we will look at the journal <em>Philosophical Transactions</em>, the longest running scientific journal in the world. We first run <code>cr_journals</code> with <code>works</code> set to <code>FALSE</code> in order to get the journal details, specifically to find out how many articles from this journal are indexed in Crossref.</p>
<pre class="r"><code>philo_issn &lt;- &quot;2053-9223&quot;
philo_details &lt;- rcrossref::cr_journals(philo_issn, works = FALSE) %&gt;% pluck(&quot;data&quot;)
philo_details$total_dois</code></pre>
<pre><code>## [1] 8534</code></pre>
<p>Because there are 8,534 articles, we need to pass the <code>cursor</code> argument to <code>cr_journals</code>, which is called “deep paging.” As described by <a href="https://poldham.github.io/abs/crossref.html">Paul Oldham</a>:</p>
<blockquote>
<p>“the CrossRef API also permits deep paging (fetching results from multiple pages). We can retrieve all the results by setting the cursor to the wildcard * and the cursor_max to the total results. Note that the cursor and the cursor_max arguments must appear together for this to work.”</p>
</blockquote>
<p>See more in the <a href="https://github.com/CrossRef/rest-api-doc#deep-paging-with-cursors">API documentation</a></p>
<p>We set <code>cursor</code> to <code>*</code> and <code>max</code> to the maximum number of results to return.</p>
<pre class="r"><code>philo_articles &lt;- rcrossref::cr_journals(philo_issn, works = TRUE, cursor = &quot;*&quot;, 
    cursor_max = 8534) %&gt;% pluck(&quot;data&quot;)</code></pre>
</div>
<div id="running-general-queries-on-cr_works" class="section level2">
<h2>Running general queries on <code>cr_works()</code></h2>
<p>You can also use <code>cr_works()</code> to run a query based on very simple text keywords. For example, you can run <code>oa_works &amp;lt;- rcrossref::cr_works(query = &quot;open+access&quot;)</code>. Paul Oldham <a href="https://poldham.github.io/abs/crossref.html#searching_crossref">gives a great example of this</a>, but does make the comment:</p>
<p>&gt; CrossRef is not a text based search engine and the ability to conduct text based searches is presently crude. Furthermore, we can only search a very limited number of fields and this will inevitably result in lower returns than commercial databases such as Web of Science (where abstracts and author keywords are available). Unfortunately there is no boolean AND for Crossref queries (see <a href="https://github.com/CrossRef/rest-api-doc/issues/135" class="uri">https://github.com/CrossRef/rest-api-doc/issues/135</a> and <a href="https://twitter.com/CrossrefSupport/status/1073601263659610113" class="uri">https://twitter.com/CrossrefSupport/status/1073601263659610113</a>). However, as discussed above, the Crossref API assigns a score to each item returned giving a measure of the API’s confidence in the match, and if you connect words using <code>+</code> the Crossref API will give items with those terms a higher score.</p>
</div>
<div id="specifying-field-queries-to-cr_works-with-flq" class="section level2">
<h2>Specifying field queries to <code>cr_works()</code> with <code>flq</code></h2>
<p>As with <code>cr_journals</code>, you can use <code>flq</code> to pass field queries on to <code>cr_works()</code>, such as author.</p>
<p>Here we search for the book <em>Open Access</em> by Peter Suber by doing a general keyword search for “open access” and an author search for “suber”:</p>
<pre class="r"><code># do a general query for the term open access and a field query to return results
# where the author name includes Suber
suber_oa &lt;- cr_works(query = &quot;open+access&quot;, flq = c(query.author = &quot;suber&quot;)) %&gt;% 
    pluck(&quot;data&quot;)

# print the data frame with select columns
suber_oa %&gt;% dplyr::select(title, doi, volume, issue, page, issued, url, publisher, 
    reference.count, type, issn)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["title"],"name":[1],"type":["chr"],"align":["left"]},{"label":["doi"],"name":[2],"type":["chr"],"align":["left"]},{"label":["volume"],"name":[3],"type":["chr"],"align":["left"]},{"label":["issue"],"name":[4],"type":["chr"],"align":["left"]},{"label":["page"],"name":[5],"type":["chr"],"align":["left"]},{"label":["issued"],"name":[6],"type":["chr"],"align":["left"]},{"label":["url"],"name":[7],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[8],"type":["chr"],"align":["left"]},{"label":["reference.count"],"name":[9],"type":["chr"],"align":["left"]},{"label":["type"],"name":[10],"type":["chr"],"align":["left"]},{"label":["issn"],"name":[11],"type":["chr"],"align":["left"]}],"data":[{"1":"Open access in the USA","2":"10.1016/b978-1-84334-203-8.50015-7","3":"NA","4":"NA","5":"149-160","6":"2006","7":"http://dx.doi.org/10.1016/b978-1-84334-203-8.50015-7","8":"Elsevier","9":"0","10":"book-chapter","11":"NA"},{"1":"Open Access","2":"10.7551/mitpress/9286.001.0001","3":"NA","4":"NA","5":"NA","6":"2012","7":"http://dx.doi.org/10.7551/mitpress/9286.001.0001","8":"The MIT Press","9":"0","10":"book","11":"NA"},{"1":"Open Access","2":"10.32388/499151","3":"NA","4":"NA","5":"NA","6":"2019-07-19","7":"http://dx.doi.org/10.32388/499151","8":"Qeios","9":"1","10":"reference-entry","11":"NA"},{"1":"Gratis Open Access","2":"10.32388/878441","3":"NA","4":"NA","5":"NA","6":"2019-07-22","7":"http://dx.doi.org/10.32388/878441","8":"Qeios","9":"1","10":"reference-entry","11":"NA"},{"1":"Knowledge Unbound","2":"10.7551/mitpress/8479.001.0001","3":"NA","4":"NA","5":"NA","6":"2016","7":"http://dx.doi.org/10.7551/mitpress/8479.001.0001","8":"The MIT Press","9":"0","10":"book","11":"NA"},{"1":"Open Access in 2008","2":"10.3998/3336451.0012.104","3":"12","4":"1","5":"NA","6":"2009-02-15","7":"http://dx.doi.org/10.3998/3336451.0012.104","8":"University of Michigan Library","9":"0","10":"journal-article","11":"1080-2711"},{"1":"Open Access in 2007","2":"10.3998/3336451.0011.110","3":"11","4":"1","5":"NA","6":"2008-01-30","7":"http://dx.doi.org/10.3998/3336451.0011.110","8":"University of Michigan Library","9":"0","10":"journal-article","11":"1080-2711"},{"1":"Open access: other ways","2":"10.1038/426015b","3":"426","4":"6962","5":"15-15","6":"2003-11","7":"http://dx.doi.org/10.1038/426015b","8":"Springer Science and Business Media LLC","9":"0","10":"journal-article","11":"0028-0836,1476-4687"},{"1":"Ensuring open access for publicly funded research","2":"10.1136/bmj.e5184","3":"345","4":"aug08 1","5":"e5184-e5184","6":"2012-08-08","7":"http://dx.doi.org/10.1136/bmj.e5184","8":"BMJ","9":"0","10":"journal-article","11":"1756-1833"},{"1":"Open access?","2":"10.1038/ngeo450","3":"2","4":"3","5":"155-155","6":"2009-03","7":"http://dx.doi.org/10.1038/ngeo450","8":"Springer Science and Business Media LLC","9":"1","10":"journal-article","11":"1752-0894,1752-0908"},{"1":"Promoting Open Access in the Humanities","2":"10.1353/syl.2005.0001","3":"16","4":"1","5":"231-246","6":"2005","7":"http://dx.doi.org/10.1353/syl.2005.0001","8":"Project Muse","9":"0","10":"journal-article","11":"2160-5157"},{"1":"Open Access to Electronic Theses and Dissertations","2":"10.14429/djlit.28.1.149","3":"28","4":"1","5":"25-34","6":"2008-01-01","7":"http://dx.doi.org/10.14429/djlit.28.1.149","8":"Defence Scientific Information and Documentation Centre","9":"0","10":"journal-article","11":"0974-0643,0976-4658"},{"1":"Open Access and Quality","2":"10.14429/djlit.28.1.152","3":"28","4":"1","5":"49-56","6":"2008-01-01","7":"http://dx.doi.org/10.14429/djlit.28.1.152","8":"Defence Scientific Information and Documentation Centre","9":"0","10":"journal-article","11":"0974-0643,0976-4658"},{"1":"Thoughts on prestige, quality, and open access","2":"10.1163/095796510x546959","3":"21","4":"1-2","5":"115-128","6":"2010","7":"http://dx.doi.org/10.1163/095796510x546959","8":"Brill","9":"0","10":"journal-article","11":"0957-9656,1878-4712"},{"1":"Open access, impact, and demand","2":"10.1136/bmj.330.7500.1097","3":"330","4":"7500","5":"1097-1098","6":"2005-05-12","7":"http://dx.doi.org/10.1136/bmj.330.7500.1097","8":"BMJ","9":"0","10":"journal-article","11":"0959-8138,1468-5833"},{"1":"SCHOLARLY COMMUNICATION: Removing barriers to research: An introduction to open access for librarians","2":"10.5860/crln.64.2.92","3":"64","4":"2","5":"92-113","6":"2003-02-01","7":"http://dx.doi.org/10.5860/crln.64.2.92","8":"American Library Association","9":"0","10":"journal-article","11":"2150-6698,0099-0086"},{"1":"A living open book","2":"10.5334/bal.l","3":"NA","4":"NA","5":"113-117","6":"2014-11-28","7":"http://dx.doi.org/10.5334/bal.l","8":"Ubiquity Press","9":"0","10":"book-chapter","11":"NA"},{"1":"Public access to federally funded research: The Cornyn-Lieberman and CURES bills","2":"10.5860/crln.67.6.7637","3":"67","4":"6","5":"362-365","6":"2006-06-01","7":"http://dx.doi.org/10.5860/crln.67.6.7637","8":"American Library Association","9":"0","10":"journal-article","11":"2150-6698,0099-0086"},{"1":"Support for the NIH Public Access Policy","2":"10.1126/science.313.5793.1572a","3":"313","4":"5793","5":"1572a-1572a","6":"2006-09-15","7":"http://dx.doi.org/10.1126/science.313.5793.1572a","8":"American Association for the Advancement of Science (AAAS)","9":"0","10":"journal-article","11":"0036-8075,1095-9203"},{"1":"NSF 19-501 AccelNet Proposal: Community of Open Scholarship Grassroots Networks (COSGN)","2":"10.31222/osf.io/d7mwk","3":"NA","4":"NA","5":"NA","6":"2020-01-30","7":"http://dx.doi.org/10.31222/osf.io/d7mwk","8":"Center for Open Science","9":"0","10":"posted-content","11":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Dr. Suber has written lots of materials that includes the term “open access.” We can use the <code>filter()</code> function from <code>dplyr</code> to look only at books, from the <strong>type</strong> column:</p>
<pre class="r"><code># use filter() from dplyr to filter that result to include only books
suber_oa_books &lt;- suber_oa %&gt;% filter(type == &quot;book&quot;)

# print the data frame with select columns
suber_oa_books %&gt;% dplyr::select(title, doi, volume, issue, page, issued, url, publisher, 
    reference.count, type, issn)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["title"],"name":[1],"type":["chr"],"align":["left"]},{"label":["doi"],"name":[2],"type":["chr"],"align":["left"]},{"label":["volume"],"name":[3],"type":["chr"],"align":["left"]},{"label":["issue"],"name":[4],"type":["chr"],"align":["left"]},{"label":["page"],"name":[5],"type":["chr"],"align":["left"]},{"label":["issued"],"name":[6],"type":["chr"],"align":["left"]},{"label":["url"],"name":[7],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[8],"type":["chr"],"align":["left"]},{"label":["reference.count"],"name":[9],"type":["chr"],"align":["left"]},{"label":["type"],"name":[10],"type":["chr"],"align":["left"]},{"label":["issn"],"name":[11],"type":["chr"],"align":["left"]}],"data":[{"1":"Open Access","2":"10.7551/mitpress/9286.001.0001","3":"NA","4":"NA","5":"NA","6":"2012","7":"http://dx.doi.org/10.7551/mitpress/9286.001.0001","8":"The MIT Press","9":"0","10":"book","11":"NA"},{"1":"Knowledge Unbound","2":"10.7551/mitpress/8479.001.0001","3":"NA","4":"NA","5":"NA","6":"2016","7":"http://dx.doi.org/10.7551/mitpress/8479.001.0001","8":"The MIT Press","9":"0","10":"book","11":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>One is the book from MIT Press that we’re looking for; the other is <em>Knowledge Unbound</em>, which is a collection of his writings.</p>
<p>We could be more specific from the outset by adding bibliographic information in <code>query.bibliographic</code>, such as ISBN (or ISSN, if it’s a journal):</p>
<pre class="r"><code># run a different cr_works() query with author set to Suber and his book&#39;s ISBN
# passed to query.bibliographic
suber_isbn &lt;- cr_works(flq = c(query.author = &quot;suber&quot;, query.bibliographic = &quot;9780262301732&quot;)) %&gt;% 
    pluck(&quot;data&quot;)

# print the data frame with select columns
suber_isbn %&gt;% dplyr::select(title, doi, issued, url, publisher, type, author)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["title"],"name":[1],"type":["chr"],"align":["left"]},{"label":["doi"],"name":[2],"type":["chr"],"align":["left"]},{"label":["issued"],"name":[3],"type":["chr"],"align":["left"]},{"label":["url"],"name":[4],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[5],"type":["chr"],"align":["left"]},{"label":["type"],"name":[6],"type":["chr"],"align":["left"]},{"label":["author"],"name":[7],"type":["list"],"align":["right"]}],"data":[{"1":"Open Access","2":"10.7551/mitpress/9286.001.0001","3":"2012","4":"http://dx.doi.org/10.7551/mitpress/9286.001.0001","5":"The MIT Press","6":"book","7":"<tibble>"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>You can combine the <code>filter</code> argument with <code>flq</code> to return only items of <strong>type</strong> <code>book</code> published in 2012.</p>
</div>
<div id="getting-formatted-references-in-a-text-file" class="section level2">
<h2>Getting formatted references in a text file</h2>
<p>We can use the <code>cr_cn()</code> function from the <code>rcrossref</code> package to get the citations to those articles in text form in the style you specify. We’ll put it into Chicago. The <code>cr_cn()</code> function returns each citation into a list element. We can use the <code>map_chr</code> and the <code>pluck</code> functions from <code>purrr</code> to instead assign them to a character vector.</p>
<pre class="r"><code># Use c() to create a vector of DOIs
my_dois &lt;- c(&quot;10.2139/ssrn.2697412&quot;, &quot;10.1016/j.joi.2016.08.002&quot;, &quot;10.1371/journal.pone.0020961&quot;, 
    &quot;10.3389/fpsyg.2018.01487&quot;, &quot;10.1038/d41586-018-00104-7&quot;, &quot;10.12688/f1000research.8460.2&quot;, 
    &quot;10.7551/mitpress/9286.001.0001&quot;)

# Use cr_cn to get back citations formatted in Chicago for those DOIs
my_citations &lt;- rcrossref::cr_cn(my_dois, format = &quot;text&quot;, style = &quot;chicago-note-bibliography&quot;) %&gt;% 
    purrr::map_chr(., purrr::pluck, 1)

# print the formatted citations
my_citations</code></pre>
<pre><code>## [1] &quot;Frosio, Giancarlo F. “Open Access Publishing: A Literature Review.” SSRN Electronic Journal (2014). doi:10.2139/ssrn.2697412.&quot;                                                                                                                                                              
## [2] &quot;Laakso, Mikael, and Bo-Christer Björk. “Hybrid Open access—A Longitudinal Study.” Journal of Informetrics 10, no. 4 (November 2016): 919–932. doi:10.1016/j.joi.2016.08.002.&quot;                                                                                                               
## [3] &quot;Laakso, Mikael, Patrik Welling, Helena Bukvova, Linus Nyman, Bo-Christer Björk, and Turid Hedlund. “The Development of Open Access Journal Publishing from 1993 to 2009.” Edited by Marcelo Hermes-Lima. PLoS ONE 6, no. 6 (June 13, 2011): e20961. doi:10.1371/journal.pone.0020961.&quot;      
## [4] &quot;Paulus, Frieder M., Nicole Cruz, and Sören Krach. “The Impact Factor Fallacy.” Frontiers in Psychology 9 (August 20, 2018). doi:10.3389/fpsyg.2018.01487.&quot;                                                                                                                                  
## [5] &quot;Shotton, David. “Funders Should Mandate Open Citations.” Nature 553, no. 7687 (January 2018): 129–129. doi:10.1038/d41586-018-00104-7.&quot;                                                                                                                                                     
## [6] &quot;Tennant, Jonathan P., François Waldner, Damien C. Jacques, Paola Masuzzo, Lauren B. Collister, and Chris. H. J. Hartgerink. “The Academic, Economic and Societal Impacts of Open Access: An Evidence-Based Review.” F1000Research 5 (June 9, 2016): 632. doi:10.12688/f1000research.8460.2.&quot;
## [7] &quot;Suber, Peter. “Open Access” (2012). doi:10.7551/mitpress/9286.001.0001.&quot;</code></pre>
<p>Beautiful formatted citations from simply a list of DOIs! You can then write this to a text file using <code>writeLines</code>.</p>
<pre class="r"><code># write the formatted citations to a text file
writeLines(my_citations, &quot;my_citations_text.txt&quot;)</code></pre>
<p>The above is helpful if you need to paste the references somewhere, and there are loads of other citation styles included in <code>rcrossref</code>–view them by calling <code>rcrossref::get_styles()</code> and it will print a vector of these styles to your console. I’ll just print the first 15 below:</p>
<pre class="r"><code># look at the first 15 styles Crossref offers
rcrossref::get_styles()[1:15]</code></pre>
<pre><code>##  [1] &quot;academy-of-management-review&quot;                                
##  [2] &quot;accident-analysis-and-prevention&quot;                            
##  [3] &quot;aci-materials-journal&quot;                                       
##  [4] &quot;acm-sig-proceedings-long-author-list&quot;                        
##  [5] &quot;acm-sig-proceedings&quot;                                         
##  [6] &quot;acm-sigchi-proceedings-extended-abstract-format&quot;             
##  [7] &quot;acm-sigchi-proceedings&quot;                                      
##  [8] &quot;acm-siggraph&quot;                                                
##  [9] &quot;acme-an-international-journal-for-critical-geographies&quot;      
## [10] &quot;acta-amazonica&quot;                                              
## [11] &quot;acta-anaesthesiologica-scandinavica&quot;                         
## [12] &quot;acta-anaesthesiologica-taiwanica&quot;                            
## [13] &quot;acta-botanica-croatica&quot;                                      
## [14] &quot;acta-chiropterologica&quot;                                       
## [15] &quot;acta-chirurgiae-orthopaedicae-et-traumatologiae-cechoslovaca&quot;</code></pre>
</div>
<div id="getting-formatted-references-in-a-bibtex-or-ris-file" class="section level2">
<h2>Getting formatted references in a BibTeX or RIS file</h2>
<p>In addition to a text file, you can also write it to BibTeX or RIS:</p>
<pre class="r"><code># Use cr_cn() to get BibTeX files for my DOIs
my_citations_bibtex &lt;- rcrossref::cr_cn(my_dois, format = &quot;bibtex&quot;) %&gt;% purrr::map_chr(., 
    purrr::pluck, 1)</code></pre>
<p>Write it to a .bib file using <code>writeLines()</code>:</p>
<pre class="r"><code># write to bibtex file
writeLines(my_citations_bibtex, &quot;my_citations_bibtex.bib&quot;)</code></pre>
</div>
<div id="getting-works-from-a-typed-citation-in-a-word-documenttext-file" class="section level2">
<h2>Getting works from a typed citation in a Word document/text file</h2>
<p>This can be helpful if you have a bibliography in a Word document or text file that you want to get into a reference management tool like Zotero. For instance, you may have written the citations in APA style and need to change to Chicago, but don’t want to rekey it all out. Or perhaps you jotted down your citations hastily and left out volume, issue, or page numbers, and you need a nice, fully-formatted citation.</p>
<p>If each citation is on its own line in your document’s bibliography, then you can probably paste the whole bibliography into an Excel spreadsheet. If it goes as planned, each citation will be in its own cell. You can then save it to a CSV file, which can then be read into R.</p>
<pre class="r"><code># read in a CSV file of citations
my_references &lt;- read.csv(&quot;data/references.txt&quot;, stringsAsFactors = FALSE)

# print the file
my_references</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["reference"],"name":[1],"type":["chr"],"align":["left"]}],"data":[{"1":"Frosio, G. F. (2014). Open Access Publishing: A Literature Review. SSRN Electronic Journal. https://doi.org/10.2139/ssrn.2697412"},{"1":"Laakso, M., & Bjork, B.-C. (2016). Hybrid open access: A longitudinal study. Journal of Informetrics, 10(4), 919-932. https://doi.org/10.1016/j.joi.2016.08.002"},{"1":"Laakso, M., Welling, P., Bukvova, H., Nyman, L., Bjork, B.-C., & Hedlund, T. (2011). The Development of Open Access Journal Publishing from 1993 to 2009. PLoS ONE, 6(6), e20961. https://doi.org/10.1371/journal.pone.0020961"},{"1":"Paulus, F. M., Cruz, N., & Krach, S. (2018). The Impact Factor Fallacy. Frontiers in Psychology, 9. https://doi.org/10.3389/fpsyg.2018.01487"},{"1":"Science, Digital; Hook, Daniel; Hahnel, Mark; Calvert, Ian (2019): The Ascent of Open Access. figshare. Journal contribution. https://doi.org/10.6084/m9.figshare.7618751"},{"1":"Shotton, D. (2018). Funders should mandate open citations. Nature, 553(7687), 129-129. https://doi.org/10.1038/d41586-018-00104-7"},{"1":"Suber, P. (2012). Open access. Cambridge, Mass: MIT Press."},{"1":"Tennant, J. P., Waldner, F., Jacques, D. C., Masuzzo, P., Collister, L. B., & Hartgerink, C. H. J. (2016). The academic, economic and societal impacts of Open Access: an evidence-based review. F1000Research, 5, 632. https://doi.org/10.12688/f1000research.8460.2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>As you can see, these are just raw citations, not divided into variables by their metadata elements (that is, with title in one column, author in another, etc.). But, we can now run a query to get precisely that from Crossref using <code>cr_works</code>. Because <code>cr_works</code> is not vectorized, we will need to build a loop using <code>map()</code> from the <code>purrr</code> package.</p>
<p>Don’t mind the technical details–it is basically saying to take each row and look it up in the Crossref search engine. Basically, this is the equivalent of copy/pasting the whole reference into the Crossref search engine. The loop will <code>print()</code> the citation before searching for it so we can keep track of where it is. We set the <code>limit</code> to 5 because if Crossref didn’t find it in the first 5 results, it’s not likely to be there at all.</p>
<pre class="r"><code># loop through the references column, using cr_works() to look the item up and
# return the top 5 hits
my_references_works_list &lt;- purrr::map(my_references$reference, function(x) {
    print(x)
    my_works &lt;- rcrossref::cr_works(query = x, limit = 5) %&gt;% purrr::pluck(&quot;data&quot;)
})</code></pre>
<pre><code>## [1] &quot;Frosio, G. F. (2014). Open Access Publishing: A Literature Review. SSRN Electronic Journal. https://doi.org/10.2139/ssrn.2697412&quot;
## [1] &quot;Laakso, M., &amp; Bjork, B.-C. (2016). Hybrid open access: A longitudinal study. Journal of Informetrics, 10(4), 919-932. https://doi.org/10.1016/j.joi.2016.08.002&quot;
## [1] &quot;Laakso, M., Welling, P., Bukvova, H., Nyman, L., Bjork, B.-C., &amp; Hedlund, T. (2011). The Development of Open Access Journal Publishing from 1993 to 2009. PLoS ONE, 6(6), e20961. https://doi.org/10.1371/journal.pone.0020961&quot;
## [1] &quot;Paulus, F. M., Cruz, N., &amp; Krach, S. (2018). The Impact Factor Fallacy. Frontiers in Psychology, 9. https://doi.org/10.3389/fpsyg.2018.01487&quot;
## [1] &quot;Science, Digital; Hook, Daniel; Hahnel, Mark; Calvert, Ian (2019): The Ascent of Open Access. figshare. Journal contribution. https://doi.org/10.6084/m9.figshare.7618751&quot;
## [1] &quot;Shotton, D. (2018). Funders should mandate open citations. Nature, 553(7687), 129-129. https://doi.org/10.1038/d41586-018-00104-7&quot;
## [1] &quot;Suber, P. (2012). Open access. Cambridge, Mass: MIT Press.&quot;
## [1] &quot;Tennant, J. P., Waldner, F., Jacques, D. C., Masuzzo, P., Collister, L. B., &amp; Hartgerink, C. H. J. (2016). The academic, economic and societal impacts of Open Access: an evidence-based review. F1000Research, 5, 632. https://doi.org/10.12688/f1000research.8460.2&quot;</code></pre>
<p>The Crossref API assigns a score to each item returned within each query, giving a measure of the API’s confidence in the match. The item with the highest score is returned first in the datasets. We can return the first result in each item in the <code>my_references_works_list</code> by using <code>map_dfr()</code>, which is like <code>map()</code> except it returns the results into a data frame rather than a list:</p>
<pre class="r"><code># for each reference looked up, get back the first result
my_references_works_df &lt;- my_references_works_list %&gt;% purrr::map_dfr(., function(x) {
    x[1, ]
})

# print the data frame with select columns
my_references_works_df %&gt;% dplyr::select(title, doi, volume, issue, page, issued, 
    url, publisher, reference.count, type, issn)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["title"],"name":[1],"type":["chr"],"align":["left"]},{"label":["doi"],"name":[2],"type":["chr"],"align":["left"]},{"label":["volume"],"name":[3],"type":["chr"],"align":["left"]},{"label":["issue"],"name":[4],"type":["chr"],"align":["left"]},{"label":["page"],"name":[5],"type":["chr"],"align":["left"]},{"label":["issued"],"name":[6],"type":["chr"],"align":["left"]},{"label":["url"],"name":[7],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[8],"type":["chr"],"align":["left"]},{"label":["reference.count"],"name":[9],"type":["chr"],"align":["left"]},{"label":["type"],"name":[10],"type":["chr"],"align":["left"]},{"label":["issn"],"name":[11],"type":["chr"],"align":["left"]}],"data":[{"1":"Open Access Publishing: A Literature Review","2":"10.2139/ssrn.2697412","3":"NA","4":"NA","5":"NA","6":"2014","7":"http://dx.doi.org/10.2139/ssrn.2697412","8":"Elsevier BV","9":"0","10":"journal-article","11":"1556-5068"},{"1":"Hybrid open access—A longitudinal study","2":"10.1016/j.joi.2016.08.002","3":"10","4":"4","5":"919-932","6":"2016-11","7":"http://dx.doi.org/10.1016/j.joi.2016.08.002","8":"Elsevier BV","9":"33","10":"journal-article","11":"1751-1577"},{"1":"The Development of Open Access Journal Publishing from 1993 to 2009","2":"10.1371/journal.pone.0020961","3":"6","4":"6","5":"e20961","6":"2011-06-13","7":"http://dx.doi.org/10.1371/journal.pone.0020961","8":"Public Library of Science (PLoS)","9":"31","10":"journal-article","11":"1932-6203"},{"1":"The Impact Factor Fallacy","2":"10.3389/fpsyg.2018.01487","3":"9","4":"NA","5":"NA","6":"2018-08-20","7":"http://dx.doi.org/10.3389/fpsyg.2018.01487","8":"Frontiers Media SA","9":"52","10":"journal-article","11":"1664-1078"},{"1":"Table S1: Digital copies of visualizations, alignments, and phylogenetic trees deposited at Figshare (also accessible under Figshare Collection doi:10.6084/m9.figshare.c.3521787)","2":"10.7717/peerj.2951/supp-3","3":"NA","4":"NA","5":"NA","6":"NA","7":"http://dx.doi.org/10.7717/peerj.2951/supp-3","8":"PeerJ","9":"0","10":"component","11":"NA"},{"1":"Funders should mandate open citations","2":"10.1038/d41586-018-00104-7","3":"553","4":"7687","5":"129-129","6":"2018-01","7":"http://dx.doi.org/10.1038/d41586-018-00104-7","8":"Springer Science and Business Media LLC","9":"0","10":"journal-article","11":"0028-0836,1476-4687"},{"1":"Open Access by Peter Suber. MIT Press, Cambridge, MA, U.S.A., 2012. 230 pp. ISBN: 978-0- 262-51763-8","2":"10.1162/leon_r_00535","3":"46","4":"2","5":"180-181","6":"2013-04","7":"http://dx.doi.org/10.1162/leon_r_00535","8":"MIT Press - Journals","9":"3","10":"journal-article","11":"0024-094X,1530-9282"},{"1":"The academic, economic and societal impacts of Open Access: an evidence-based review","2":"10.12688/f1000research.8460.1","3":"5","4":"NA","5":"632","6":"2016-04-11","7":"http://dx.doi.org/10.12688/f1000research.8460.1","8":"F1000 Research Ltd","9":"137","10":"journal-article","11":"2046-1402"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>We can print just the titles to quickly see how well they match with the titles of the works we requested:</p>
<pre class="r"><code># print the title column
my_references_works_df$title</code></pre>
<pre><code>## [1] &quot;Open Access Publishing: A Literature Review&quot;                                                                                                                                       
## [2] &quot;Hybrid open access—A longitudinal study&quot;                                                                                                                                           
## [3] &quot;The Development of Open Access Journal Publishing from 1993 to 2009&quot;                                                                                                               
## [4] &quot;The Impact Factor Fallacy&quot;                                                                                                                                                         
## [5] &quot;Table S1: Digital copies of visualizations, alignments, and phylogenetic trees deposited at Figshare (also accessible under Figshare Collection doi:10.6084/m9.figshare.c.3521787)&quot;
## [6] &quot;Funders should mandate open citations&quot;                                                                                                                                             
## [7] &quot;Open Access by Peter Suber. MIT Press, Cambridge, MA, U.S.A., 2012. 230 pp. ISBN: 978-0- 262-51763-8&quot;                                                                              
## [8] &quot;The academic, economic and societal impacts of Open Access: an evidence-based review&quot;</code></pre>
<p>Not bad! Not bad! Looks like we got 6 out of 8, with problems on number 5 and 7.</p>
<p>Let’s deal with 5 first. This was the result for “The Ascent of Open Access”, which was a report by Digital Science posted to figshare, didn’t come back. Even though this report does have a DOI (<a href="https://doi.org/10.6084/m9.figshare.7618751.v2" class="uri">https://doi.org/10.6084/m9.figshare.7618751.v2</a>) assigned via figshare, the <code>cr_works()</code> function searches only for Crossref DOIs. We should check to see if it came back in any of the 5 items we pulled. We do this by calling <code>pluck()</code> on the titles of the fifth item in the list:</p>
<pre class="r"><code>my_references_works_list %&gt;% purrr::pluck(5, &quot;title&quot;)</code></pre>
<pre><code>## [1] &quot;Table S1: Digital copies of visualizations, alignments, and phylogenetic trees deposited at Figshare (also accessible under Figshare Collection doi:10.6084/m9.figshare.c.3521787)&quot;                                          
## [2] &quot;Figure 4 from: Miko I, Ernst A, Deans A (2013) Morphology and function of the ovipositor mechanism in Ceraphronoidea (Hymenoptera, Apocrita). Journal of Hymenoptera Research 33: 25-61. https://doi.org/10.3897/jhr.33.5204&quot;
## [3] &quot;Figure 1 from: Miko I, Ernst A, Deans A (2013) Morphology and function of the ovipositor mechanism in Ceraphronoidea (Hymenoptera, Apocrita). Journal of Hymenoptera Research 33: 25-61. https://doi.org/10.3897/jhr.33.5204&quot;
## [4] &quot;Figure 6 from: Miko I, Popovici O, Seltmann K, Deans A (2014) The maxillo-labial complex of Sparasion (Hymenoptera, Platygastroidea). Journal of Hymenoptera Research 37: 77-111. https://doi.org/10.3897/jhr.37.5206&quot;       
## [5] &quot;Figure 3 from: Miko I, Ernst A, Deans A (2013) Morphology and function of the ovipositor mechanism in Ceraphronoidea (Hymenoptera, Apocrita). Journal of Hymenoptera Research 33: 25-61. https://doi.org/10.3897/jhr.33.5204&quot;</code></pre>
<p>Nope, unfortunately none of these are “The Ascent of Open Access”, so we’re out of luck. We can just throw this row out entirely using <code>slice()</code> from <code>dplyr</code>. We’ll overwrite our existing <code>my_references_works_df</code> because we have no future use for it in this R session.</p>
<pre class="r"><code>my_references_works_df &lt;- my_references_works_df %&gt;% dplyr::slice(-5)</code></pre>
<p>For row 7, it’s giving us the full citation for Peter Suber’s book when we asked for the title only, so something is fishy.</p>
<p>When we look at it more closely (we can call <code>View(my_references_works_df)</code>), we see the author of this item is not Peter Suber, but Rob Harle, and, checking the <strong>type</strong> column, it’s a journal article, not a book. This is a book review published in the journal <em>Leonardo</em>, not Peter Suber’s book. So let’s go back to <code>my_references_works_list</code> and pull data from all 5 items that came back with the API call and see if Suber’s book is in there somewhere:</p>
<pre class="r"><code>suber &lt;- my_references_works_list %&gt;% purrr::pluck(7)
suber</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["alternative.id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["container.title"],"name":[2],"type":["chr"],"align":["left"]},{"label":["created"],"name":[3],"type":["chr"],"align":["left"]},{"label":["deposited"],"name":[4],"type":["chr"],"align":["left"]},{"label":["published.print"],"name":[5],"type":["chr"],"align":["left"]},{"label":["doi"],"name":[6],"type":["chr"],"align":["left"]},{"label":["indexed"],"name":[7],"type":["chr"],"align":["left"]},{"label":["issn"],"name":[8],"type":["chr"],"align":["left"]},{"label":["issue"],"name":[9],"type":["chr"],"align":["left"]},{"label":["issued"],"name":[10],"type":["chr"],"align":["left"]},{"label":["member"],"name":[11],"type":["chr"],"align":["left"]},{"label":["page"],"name":[12],"type":["chr"],"align":["left"]},{"label":["prefix"],"name":[13],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[14],"type":["chr"],"align":["left"]},{"label":["reference.count"],"name":[15],"type":["chr"],"align":["left"]},{"label":["score"],"name":[16],"type":["chr"],"align":["left"]},{"label":["source"],"name":[17],"type":["chr"],"align":["left"]},{"label":["title"],"name":[18],"type":["chr"],"align":["left"]},{"label":["type"],"name":[19],"type":["chr"],"align":["left"]},{"label":["url"],"name":[20],"type":["chr"],"align":["left"]},{"label":["volume"],"name":[21],"type":["chr"],"align":["left"]},{"label":["author"],"name":[22],"type":["list"],"align":["right"]},{"label":["link"],"name":[23],"type":["list"],"align":["right"]},{"label":["reference"],"name":[24],"type":["list"],"align":["right"]},{"label":["isbn"],"name":[25],"type":["chr"],"align":["left"]},{"label":["license"],"name":[26],"type":["list"],"align":["right"]},{"label":["published.online"],"name":[27],"type":["chr"],"align":["left"]}],"data":[{"1":"10.1162/LEON_r_00535","2":"Leonardo","3":"2013-01-17","4":"2016-12-06","5":"2013-04","6":"10.1162/leon_r_00535","7":"2020-03-28","8":"0024-094X,1530-9282","9":"2","10":"2013-04","11":"281","12":"180-181","13":"10.1162","14":"MIT Press - Journals","15":"3","16":"41.006508","17":"Crossref","18":"Open Access by Peter Suber. MIT Press, Cambridge, MA, U.S.A., 2012. 230 pp. ISBN: 978-0- 262-51763-8","19":"journal-article","20":"http://dx.doi.org/10.1162/leon_r_00535","21":"46","22":"<tibble>","23":"<tibble>","24":"<tibble>","25":"NA","26":"<NULL>","27":"NA"},{"1":"NA","2":"NA","3":"2019-01-11","4":"2019-01-11","5":"2012","6":"10.7551/mitpress/9286.001.0001","7":"2020-08-01","8":"NA","9":"NA","10":"2012","11":"281","12":"NA","13":"10.7551","14":"The MIT Press","15":"0","16":"39.058643","17":"Crossref","18":"Open Access","19":"book","20":"http://dx.doi.org/10.7551/mitpress/9286.001.0001","21":"NA","22":"<tibble>","23":"<NULL>","24":"<NULL>","25":"9780262301732","26":"<NULL>","27":"NA"},{"1":"10.1108/14684521311311711","2":"Online Information Review","3":"2014-11-16","4":"2019-09-03","5":"2013-02-15","6":"10.1108/14684521311311711","7":"2020-04-10","8":"1468-4527","9":"1","10":"2013-02-15","11":"140","12":"150-151","13":"10.1108","14":"Emerald","15":"0","16":"38.816944","17":"Crossref","18":"Open Access20135Peter Suber. Open Access. Cambridge, MA: MIT Press 2012. 242 pp. US$12.95 soft cover, ISBN: 9780262517638 The MIT Press Essential Knowledge Series","19":"journal-article","20":"http://dx.doi.org/10.1108/14684521311311711","21":"37","22":"<tibble>","23":"<tibble>","24":"<NULL>","25":"NA","26":"<tibble>","27":"NA"},{"1":"NA","2":"Definitions","3":"2019-07-19","4":"2019-11-11","5":"NA","6":"10.32388/499151","7":"2020-03-26","8":"NA","9":"NA","10":"2019-07-19","11":"17262","12":"NA","13":"10.32388","14":"Qeios","15":"1","16":"34.13484","17":"Crossref","18":"Open Access","19":"reference-entry","20":"http://dx.doi.org/10.32388/499151","21":"NA","22":"<tibble>","23":"<NULL>","24":"<tibble>","25":"NA","26":"<NULL>","27":"2019-07-19"},{"1":"NA","2":"Definitions","3":"2019-07-22","4":"2019-11-11","5":"NA","6":"10.32388/878441","7":"2020-03-26","8":"NA","9":"NA","10":"2019-07-22","11":"17262","12":"NA","13":"10.32388","14":"Qeios","15":"1","16":"34.13484","17":"Crossref","18":"Gratis Open Access","19":"reference-entry","20":"http://dx.doi.org/10.32388/878441","21":"NA","22":"<tibble>","23":"<NULL>","24":"<tibble>","25":"NA","26":"<NULL>","27":"2019-07-22"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>It looks like it is the second item, confirming by seeing the <strong>author</strong> is Peter Suber, the <strong>publisher</strong> is MIT Press, the <strong>type</strong> is book, and the <strong>ISBN</strong> is “9780262301732”.</p>
<p>We do the following to correct it:</p>
<ul>
<li>use <code>filter()</code> with the isbn to assign the correct row from <code>suber</code> to a variable called <code>suber_correct</code></li>
<li>remove the incorrect row with <code>slice</code> (double checking that it is the 6th row)</li>
<li>use <code>bind_rows()</code> to add the correct one to our <code>my_references_works_df</code> data frame. We can just overwrite the existing <code>my_references_works_df</code> again</li>
</ul>
<pre class="r"><code>suber_correct &lt;- suber %&gt;% dplyr::filter(isbn == &quot;9780262301732&quot;)
my_references_works_df &lt;- my_references_works_df %&gt;% dplyr::slice(-6) %&gt;% bind_rows(suber_correct)</code></pre>
</div>
<div id="writing-publications-to-disk" class="section level2">
<h2>Writing publications to disk</h2>
<p>We will use the <code>write_csv()</code> function from the <code>readr</code> package to write our data to disk. This package was loaded when you called <code>library(tidyverse)</code> at the beginning of the session.</p>
<p>First we’re going to create a vector that represents file path to where you want to save the file:</p>
<pre class="r"><code>my_filepath &lt;- &quot;C:/EXAMPLE/crossref/data&quot;</code></pre>
<p>Unfortunately, you cannot simply write the <code>jah_jama_publications</code> data frame to a CSV, due to the nested lists. It will throw an error: <code>&quot;Error in stream_delim_(df, path, ...) : Don't know how to handle vector of type list.&quot;</code></p>
<p>You have a few choices here:</p>
<ol style="list-style-type: decimal">
<li>You can unnest one of the columns and leave <code>.drop</code> set to <code>TRUE</code>. This will add rows for all the values in the nested lists, and drop the additional nested lists.</li>
</ol>
<pre class="r"><code>jah_jama_publications_auth &lt;- jah_jama_publications %&gt;% dplyr::filter(!purrr::map_lgl(author, 
    is.null)) %&gt;% tidyr::unnest(author, .drop = TRUE) %&gt;% dplyr::bind_rows(jah_jama_publications %&gt;% 
    dplyr::filter(map_lgl(author, is.null)) %&gt;% dplyr::select(-author, -link))
readr::write_csv(jah_jama_publications_auth, file.path(my_filepath, &quot;jah_jama_publications_auth.csv&quot;))</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>You can drop the nested lists altogether using a combination of <code>select_if()</code> from <code>dplyr</code> and <code>negate()</code> from <code>purrr</code> to drop all lists in the data frame. This only works if you don’t need author or link.</li>
</ol>
<pre class="r"><code>jah_jama_publications_short &lt;- jah_jama_publications %&gt;% dplyr::select_if(purrr::negate(is.list))
readr::write_csv(jah_jama_publications_author, file.path(my_filepath, &quot;jah_jama_publications_short.csv&quot;))</code></pre>
<p>We go from 24 to 22 variables because it dropped <strong>author</strong> and <strong>list</strong>.</p>
<ol start="3" style="list-style-type: decimal">
<li>You can use <code>mutate()</code> from <code>dplyr</code> to coerce the list columns into character vectors:</li>
</ol>
<pre class="r"><code>jah_jama_publications_mutated &lt;- jah_jama_publications %&gt;% dplyr::mutate(author = as.character(author)) %&gt;% 
    dplyr::mutate(link = as.character(link))
readr::write_csv(jah_jama_publications_mutated, file.path(my_filepath, &quot;jah_jama_publications_mutated.csv&quot;))</code></pre>
</div>
</div>
<div id="using-roadoi-to-check-for-open-access" class="section level1">
<h1>Using <code>roadoi</code> to check for open access</h1>
<p><code>roadoi</code> was developed by Najko Jahn, with reviews from Tuija Sonkkila and Ross Mounce. It interfaces with <a href="https://unpaywall.org">Unpaywall</a> (which used to be called oaDOI), an important tool developed by <a href="http://unpaywall.org/team">ImpactStory</a> (Heather Piwowar and Jason Priem) for locating open access versions of scholarship–read more in this <a href="https://www.nature.com/articles/d41586-018-05968-3"><em>Nature</em> article</a>. See here for <a href="https://cran.r-project.org/web/packages/roadoi/roadoi.pdf">the <code>roadoi</code> documentation</a>.</p>
<p>This incredible <a href="https://cran.r-project.org/web/packages/roadoi/vignettes/intro.html">Introduction to <code>roadoi</code></a> by Najko Jahn provides much of what you need to know to use the tool, as well as an interesting use case. Also see his recently published article <a href="https://subugoe.github.io/scholcomm_analytics/posts/unpaywall_evidence/">Open Access Evidence in Unpaywall</a>, running deep analysis on Unpaywall data.</p>
<p>First install the package and load it.</p>
<pre class="r"><code># install the roadoi package
install.packages(&quot;roadoi&quot;)</code></pre>
<pre class="r"><code># load the roadoi package
library(roadoi)</code></pre>
<div id="setting-up-roadoi" class="section level2">
<h2>Setting up <code>roadoi</code></h2>
<p>Your API calls to Unpaywall must include a valid email address where you can be reached in order to keep the service open and free for everyone.</p>
</div>
<div id="checking-oa-status-with-oadoi_fetch" class="section level2">
<h2>Checking OA status with <code>oadoi_fetch</code></h2>
<p>We then create DOI vector and use the <code>oadoi_fetch()</code> function from <code>roadoi</code>.</p>
<p><strong>Be sure to replace the email below with your own</strong></p>
<pre class="r"><code># Use c() to create a vector of DOIs
my_dois &lt;- c(&quot;10.2139/ssrn.2697412&quot;, &quot;10.1016/j.joi.2016.08.002&quot;, &quot;10.1371/journal.pone.0020961&quot;, 
    &quot;10.3389/fpsyg.2018.01487&quot;, &quot;10.1038/d41586-018-00104-7&quot;, &quot;10.12688/f1000research.8460.2&quot;, 
    &quot;10.7551/mitpress/9286.001.0001&quot;)

# use oadoi_fetch() to get Unpaywall data on those DOIs
my_dois_oa &lt;- roadoi::oadoi_fetch(dois = my_dois, email = &quot;clarke.iakovakis@okstate.edu&quot;)</code></pre>
<p>Look at the column names.</p>
<pre class="r"><code># print column names
names(my_dois_oa)</code></pre>
<pre><code>##  [1] &quot;doi&quot;                 &quot;best_oa_location&quot;    &quot;oa_locations&quot;       
##  [4] &quot;data_standard&quot;       &quot;is_oa&quot;               &quot;genre&quot;              
##  [7] &quot;oa_status&quot;           &quot;has_repository_copy&quot; &quot;journal_is_oa&quot;      
## [10] &quot;journal_is_in_doaj&quot;  &quot;journal_issns&quot;       &quot;journal_issn_l&quot;     
## [13] &quot;journal_name&quot;        &quot;publisher&quot;           &quot;title&quot;              
## [16] &quot;year&quot;                &quot;updated&quot;             &quot;authors&quot;</code></pre>
<p>The returned variables are described on the <a href="http://unpaywall.org/data-format">Unpaywall Data Format</a> page.</p>
<p>We can see that Unpaywall could not find OA versions for two of the seven of these, so we will filter them out:</p>
<pre class="r"><code># use filter() to overwrite the data frame and keep only items that are available
# OA
my_dois_oa &lt;- my_dois_oa %&gt;% dplyr::filter(is_oa == TRUE)</code></pre>
<p>As above, it is easier to use <code>unnest()</code> to more closely view one of the variables:</p>
<pre class="r"><code># print the data frame with best open access location unnested
my_dois_oa %&gt;% tidyr::unnest(best_oa_location, names_repair = &quot;universal&quot;)</code></pre>
<pre><code>## New names:
## * updated -&gt; updated...9
## * updated -&gt; updated...28</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["doi"],"name":[1],"type":["chr"],"align":["left"]},{"label":["endpoint_id"],"name":[2],"type":["chr"],"align":["left"]},{"label":["evidence"],"name":[3],"type":["chr"],"align":["left"]},{"label":["host_type"],"name":[4],"type":["chr"],"align":["left"]},{"label":["is_best"],"name":[5],"type":["lgl"],"align":["right"]},{"label":["license"],"name":[6],"type":["chr"],"align":["left"]},{"label":["pmh_id"],"name":[7],"type":["chr"],"align":["left"]},{"label":["repository_institution"],"name":[8],"type":["chr"],"align":["left"]},{"label":["updated...9"],"name":[9],"type":["chr"],"align":["left"]},{"label":["url"],"name":[10],"type":["chr"],"align":["left"]},{"label":["url_for_landing_page"],"name":[11],"type":["chr"],"align":["left"]},{"label":["url_for_pdf"],"name":[12],"type":["chr"],"align":["left"]},{"label":["version"],"name":[13],"type":["chr"],"align":["left"]},{"label":["oa_locations"],"name":[14],"type":["list"],"align":["right"]},{"label":["data_standard"],"name":[15],"type":["int"],"align":["right"]},{"label":["is_oa"],"name":[16],"type":["lgl"],"align":["right"]},{"label":["genre"],"name":[17],"type":["chr"],"align":["left"]},{"label":["oa_status"],"name":[18],"type":["chr"],"align":["left"]},{"label":["has_repository_copy"],"name":[19],"type":["lgl"],"align":["right"]},{"label":["journal_is_oa"],"name":[20],"type":["lgl"],"align":["right"]},{"label":["journal_is_in_doaj"],"name":[21],"type":["lgl"],"align":["right"]},{"label":["journal_issns"],"name":[22],"type":["chr"],"align":["left"]},{"label":["journal_issn_l"],"name":[23],"type":["chr"],"align":["left"]},{"label":["journal_name"],"name":[24],"type":["chr"],"align":["left"]},{"label":["publisher"],"name":[25],"type":["chr"],"align":["left"]},{"label":["title"],"name":[26],"type":["chr"],"align":["left"]},{"label":["year"],"name":[27],"type":["chr"],"align":["left"]},{"label":["updated...28"],"name":[28],"type":["chr"],"align":["left"]},{"label":["authors"],"name":[29],"type":["list"],"align":["right"]}],"data":[{"1":"10.2139/ssrn.2697412","2":"kmlgamtfmx4w6tf78ykl","3":"oa repository (via OAI-PMH title and first author match)","4":"repository","5":"TRUE","6":"cc-by","7":"oai:zenodo.org:8381","8":"CERN European Organization for Nuclear Research - Zenodo","9":"2020-08-01T22:09:50.925791","10":"https://zenodo.org/record/8381/files/CREATe-Working-Paper-2014-01.pdf","11":"https://zenodo.org/record/8381","12":"https://zenodo.org/record/8381/files/CREATe-Working-Paper-2014-01.pdf","13":"publishedVersion","14":"<tibble>","15":"2","16":"TRUE","17":"journal-article","18":"green","19":"TRUE","20":"FALSE","21":"FALSE","22":"1556-5068","23":"1556-5068","24":"SSRN Electronic Journal","25":"Elsevier BV","26":"Open Access Publishing: A Literature Review","27":"2014","28":"2020-07-01T13:56:47.528528","29":"<data.frame [1 × 3]>"},{"1":"10.1016/j.joi.2016.08.002","2":"NA","3":"open (via crossref license)","4":"publisher","5":"TRUE","6":"cc-by","7":"NA","8":"NA","9":"2020-08-07T13:20:38.154215","10":"https://doi.org/10.1016/j.joi.2016.08.002","11":"https://doi.org/10.1016/j.joi.2016.08.002","12":"NA","13":"publishedVersion","14":"<tibble>","15":"2","16":"TRUE","17":"journal-article","18":"hybrid","19":"FALSE","20":"FALSE","21":"FALSE","22":"1751-1577","23":"1751-1577","24":"Journal of Informetrics","25":"Elsevier BV","26":"Hybrid open access—A longitudinal study","27":"2016","28":"2020-03-14T19:55:45.992330","29":"<data.frame [2 × 5]>"},{"1":"10.1371/journal.pone.0020961","2":"NA","3":"open (via page says license)","4":"publisher","5":"TRUE","6":"cc-by","7":"NA","8":"NA","9":"2018-09-26T18:32:55.188635","10":"https://journals.plos.org/plosone/article/file?id=10.1371/journal.pone.0020961&type=printable","11":"https://doi.org/10.1371/journal.pone.0020961","12":"https://journals.plos.org/plosone/article/file?id=10.1371/journal.pone.0020961&type=printable","13":"publishedVersion","14":"<tibble>","15":"2","16":"TRUE","17":"journal-article","18":"gold","19":"TRUE","20":"TRUE","21":"TRUE","22":"1932-6203","23":"1932-6203","24":"PLoS ONE","25":"Public Library of Science (PLoS)","26":"The Development of Open Access Journal Publishing from 1993 to 2009","27":"2011","28":"2020-07-13T01:19:02.393223","29":"<data.frame [6 × 3]>"},{"1":"10.3389/fpsyg.2018.01487","2":"NA","3":"open (via page says license)","4":"publisher","5":"TRUE","6":"cc-by","7":"NA","8":"NA","9":"2019-11-09T20:20:21.096918","10":"https://www.frontiersin.org/articles/10.3389/fpsyg.2018.01487/pdf","11":"https://doi.org/10.3389/fpsyg.2018.01487","12":"https://www.frontiersin.org/articles/10.3389/fpsyg.2018.01487/pdf","13":"publishedVersion","14":"<tibble>","15":"2","16":"TRUE","17":"journal-article","18":"gold","19":"TRUE","20":"TRUE","21":"TRUE","22":"1664-1078","23":"1664-1078","24":"Frontiers in Psychology","25":"Frontiers Media SA","26":"The Impact Factor Fallacy","27":"2018","28":"2020-07-25T07:21:45.359716","29":"<data.frame [3 × 3]>"},{"1":"10.1038/d41586-018-00104-7","2":"NA","3":"open (via free pdf)","4":"publisher","5":"TRUE","6":"pd","7":"NA","8":"NA","9":"2019-12-23T19:38:29.249435","10":"https://www.nature.com/magazine-assets/d41586-018-00104-7/d41586-018-00104-7.pdf","11":"https://doi.org/10.1038/d41586-018-00104-7","12":"https://www.nature.com/magazine-assets/d41586-018-00104-7/d41586-018-00104-7.pdf","13":"publishedVersion","14":"<tibble>","15":"2","16":"TRUE","17":"journal-article","18":"hybrid","19":"FALSE","20":"FALSE","21":"FALSE","22":"0028-0836,1476-4687","23":"0028-0836","24":"Nature","25":"Springer Science and Business Media LLC","26":"Funders should mandate open citations","27":"2018","28":"2020-02-27T21:47:08.667168","29":"<data.frame [1 × 3]>"},{"1":"10.12688/f1000research.8460.2","2":"NA","3":"open (via page says license)","4":"publisher","5":"TRUE","6":"cc-by","7":"NA","8":"NA","9":"2020-01-18T08:07:52.322656","10":"https://f1000research.com/articles/5-632/v2/pdf","11":"https://doi.org/10.12688/f1000research.8460.2","12":"https://f1000research.com/articles/5-632/v2/pdf","13":"publishedVersion","14":"<tibble>","15":"2","16":"TRUE","17":"journal-article","18":"gold","19":"TRUE","20":"TRUE","21":"TRUE","22":"2046-1402","23":"2046-1402","24":"F1000Research","25":"F1000 Research Ltd","26":"The academic, economic and societal impacts of Open Access: an evidence-based review","27":"2016","28":"2020-06-28T10:25:20.453064","29":"<data.frame [6 × 5]>"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="next-steps" class="section level1">
<h1>Next steps</h1>
<p>There are several other excellent R packages that interface with publication metadata that can be used in conjunction with this package. Examples:</p>
<ul>
<li><code>bibliometrix</code> “is an open-source tool for quantitative research in scientometrics and bibliometrics that includes all the main bibliometric methods of analysis.” See more information at <a href="https://bibliometrix.org/" class="uri">https://bibliometrix.org/</a>.</li>
<li><code>rromeo</code> is a wrapper for the SHERPA-RoMEO API. You can retrieve a set of publications metadata from <code>rcrossref</code>, then use the ISSN to look up the policies of the journal regarding the archival of preprints, postprints, and publisher versions. <a href="https://cran.r-project.org/web/packages/rromeo/rromeo.pdf" class="uri">https://cran.r-project.org/web/packages/rromeo/rromeo.pdf</a></li>
<li><code>crminer</code> “includes functions for getting getting links to full text of articles, fetching full text articles from those links or Digital Object Identifiers (‘DOIs’), and text extraction from ‘PDFs’.” <a href="https://cran.r-project.org/web/packages/crminer/crminer.pdf" class="uri">https://cran.r-project.org/web/packages/crminer/crminer.pdf</a></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
